// @openrouter/sdk/esm/models/operations/sendchatcompletionrequest.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 52a55a780d2d
 */
import * as z from "zod/v4";
import { EventStream } from "../../lib/event-streams.js";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import * as models from "../index.js";
/** @internal */
export const SendChatCompletionRequestRequest$outboundSchema = z.object({
    httpReferer: z.string().optional(),
    appTitle: z.string().optional(),
    appCategories: z.string().optional(),
    chatGenerationParams: models.ChatGenerationParams$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        httpReferer: "HTTP-Referer",
        chatGenerationParams: "ChatGenerationParams",
    });
});
export function sendChatCompletionRequestRequestToJSON(sendChatCompletionRequestRequest) {
    return JSON.stringify(SendChatCompletionRequestRequest$outboundSchema.parse(sendChatCompletionRequestRequest));
}
/** @internal */
export const SendChatCompletionRequestResponseBody$inboundSchema = z.object({
    data: z.string().transform((v, ctx) => {
        try {
            return JSON.parse(v);
        }
        catch (err) {
            ctx.addIssue({
                input: v,
                code: "custom",
                message: `malformed json: ${err}`,
            });
            return z.NEVER;
        }
    }).pipe(models.ChatStreamingResponseChunk$inboundSchema),
});
export function sendChatCompletionRequestResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SendChatCompletionRequestResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SendChatCompletionRequestResponseBody' from JSON`);
}
/** @internal */
export const SendChatCompletionRequestResponse$inboundSchema = z.union([
    models.ChatResponse$inboundSchema,
    z.custom(x => x instanceof ReadableStream)
        .transform(stream => {
        return new EventStream(stream, rawEvent => {
            if (rawEvent.data === "[DONE]")
                return { done: true };
            return {
                value: z.lazy(() => SendChatCompletionRequestResponseBody$inboundSchema).parse(rawEvent)?.data,
            };
        });
    }),
]);
export function sendChatCompletionRequestResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SendChatCompletionRequestResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SendChatCompletionRequestResponse' from JSON`);
}
