// @openrouter/sdk/esm/models/assistantmessageimages.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 7b893584cc21
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const AssistantMessageImagesImageUrl$inboundSchema = z.object({
    url: z.string(),
});
/** @internal */
export const AssistantMessageImagesImageUrl$outboundSchema = z.object({
    url: z.string(),
});
export function assistantMessageImagesImageUrlToJSON(assistantMessageImagesImageUrl) {
    return JSON.stringify(AssistantMessageImagesImageUrl$outboundSchema.parse(assistantMessageImagesImageUrl));
}
export function assistantMessageImagesImageUrlFromJSON(jsonString) {
    return safeParse(jsonString, (x) => AssistantMessageImagesImageUrl$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AssistantMessageImagesImageUrl' from JSON`);
}
/** @internal */
export const AssistantMessageImages$inboundSchema = z.object({
    image_url: z.lazy(() => AssistantMessageImagesImageUrl$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "image_url": "imageUrl",
    });
});
/** @internal */
export const AssistantMessageImages$outboundSchema = z.object({
    imageUrl: z.lazy(() => AssistantMessageImagesImageUrl$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        imageUrl: "image_url",
    });
});
export function assistantMessageImagesToJSON(assistantMessageImages) {
    return JSON.stringify(AssistantMessageImages$outboundSchema.parse(assistantMessageImages));
}
export function assistantMessageImagesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => AssistantMessageImages$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AssistantMessageImages' from JSON`);
}
