// @openrouter/sdk/esm/models/assistantmessage.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 2147a953e992
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { AssistantMessageImages$inboundSchema, AssistantMessageImages$outboundSchema, } from "./assistantmessageimages.js";
import { ChatCompletionAudioOutput$inboundSchema, ChatCompletionAudioOutput$outboundSchema, } from "./chatcompletionaudiooutput.js";
import { ChatMessageContentItem$inboundSchema, ChatMessageContentItem$outboundSchema, } from "./chatmessagecontentitem.js";
import { ChatMessageToolCall$inboundSchema, ChatMessageToolCall$outboundSchema, } from "./chatmessagetoolcall.js";
import { ReasoningDetailUnion$inboundSchema, ReasoningDetailUnion$outboundSchema, } from "./reasoningdetailunion.js";
export const AssistantMessageRole = {
    Assistant: "assistant",
};
/** @internal */
export const AssistantMessageRole$inboundSchema = z.enum(AssistantMessageRole);
/** @internal */
export const AssistantMessageRole$outboundSchema = AssistantMessageRole$inboundSchema;
/** @internal */
export const AssistantMessageContent$inboundSchema = z.union([
    z.string(),
    z.array(ChatMessageContentItem$inboundSchema),
    z.any(),
]);
/** @internal */
export const AssistantMessageContent$outboundSchema = z.union([
    z.string(),
    z.array(ChatMessageContentItem$outboundSchema),
    z.any(),
]);
export function assistantMessageContentToJSON(assistantMessageContent) {
    return JSON.stringify(AssistantMessageContent$outboundSchema.parse(assistantMessageContent));
}
export function assistantMessageContentFromJSON(jsonString) {
    return safeParse(jsonString, (x) => AssistantMessageContent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AssistantMessageContent' from JSON`);
}
/** @internal */
export const AssistantMessage$inboundSchema = z.object({
    role: AssistantMessageRole$inboundSchema,
    content: z.nullable(z.union([
        z.string(),
        z.array(ChatMessageContentItem$inboundSchema),
        z.any(),
    ])).optional(),
    name: z.string().optional(),
    tool_calls: z.array(ChatMessageToolCall$inboundSchema).optional(),
    refusal: z.nullable(z.string()).optional(),
    reasoning: z.nullable(z.string()).optional(),
    reasoning_details: z.array(ReasoningDetailUnion$inboundSchema).optional(),
    images: z.array(AssistantMessageImages$inboundSchema).optional(),
    audio: ChatCompletionAudioOutput$inboundSchema.optional(),
}).transform((v) => {
    return remap$(v, {
        "tool_calls": "toolCalls",
        "reasoning_details": "reasoningDetails",
    });
});
/** @internal */
export const AssistantMessage$outboundSchema = z.object({
    role: AssistantMessageRole$outboundSchema,
    content: z.nullable(z.union([
        z.string(),
        z.array(ChatMessageContentItem$outboundSchema),
        z.any(),
    ])).optional(),
    name: z.string().optional(),
    toolCalls: z.array(ChatMessageToolCall$outboundSchema).optional(),
    refusal: z.nullable(z.string()).optional(),
    reasoning: z.nullable(z.string()).optional(),
    reasoningDetails: z.array(ReasoningDetailUnion$outboundSchema).optional(),
    images: z.array(AssistantMessageImages$outboundSchema).optional(),
    audio: ChatCompletionAudioOutput$outboundSchema.optional(),
}).transform((v) => {
    return remap$(v, {
        toolCalls: "tool_calls",
        reasoningDetails: "reasoning_details",
    });
});
export function assistantMessageToJSON(assistantMessage) {
    return JSON.stringify(AssistantMessage$outboundSchema.parse(assistantMessage));
}
export function assistantMessageFromJSON(jsonString) {
    return safeParse(jsonString, (x) => AssistantMessage$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AssistantMessage' from JSON`);
}
