// @openrouter/sdk/esm/models/chatmessagecontentitemvideolegacy.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 785ee29d8430
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { VideoInput$inboundSchema, VideoInput$outboundSchema, } from "./videoinput.js";
/** @internal */
export const ChatMessageContentItemVideoLegacy$inboundSchema = z.object({
    type: z.literal("input_video"),
    video_url: VideoInput$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "video_url": "videoUrl",
    });
});
/** @internal */
export const ChatMessageContentItemVideoLegacy$outboundSchema = z.object({
    type: z.literal("input_video"),
    videoUrl: VideoInput$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        videoUrl: "video_url",
    });
});
export function chatMessageContentItemVideoLegacyToJSON(chatMessageContentItemVideoLegacy) {
    return JSON.stringify(ChatMessageContentItemVideoLegacy$outboundSchema.parse(chatMessageContentItemVideoLegacy));
}
export function chatMessageContentItemVideoLegacyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatMessageContentItemVideoLegacy$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatMessageContentItemVideoLegacy' from JSON`);
}
