// @openrouter/sdk/esm/models/chatmessagecontentitemfile.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: ec73b7087290
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
export const ChatMessageContentItemFileType = {
    File: "file",
};
/** @internal */
export const ChatMessageContentItemFileType$inboundSchema = z.enum(ChatMessageContentItemFileType);
/** @internal */
export const ChatMessageContentItemFileType$outboundSchema = ChatMessageContentItemFileType$inboundSchema;
/** @internal */
export const FileT$inboundSchema = z.object({
    file_data: z.string().optional(),
    file_id: z.string().optional(),
    filename: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        "file_data": "fileData",
        "file_id": "fileId",
    });
});
/** @internal */
export const FileT$outboundSchema = z.object({
    fileData: z.string().optional(),
    fileId: z.string().optional(),
    filename: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        fileData: "file_data",
        fileId: "file_id",
    });
});
export function fileToJSON(fileT) {
    return JSON.stringify(FileT$outboundSchema.parse(fileT));
}
export function fileFromJSON(jsonString) {
    return safeParse(jsonString, (x) => FileT$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FileT' from JSON`);
}
/** @internal */
export const ChatMessageContentItemFile$inboundSchema = z.object({
    type: ChatMessageContentItemFileType$inboundSchema,
    file: z.lazy(() => FileT$inboundSchema),
});
/** @internal */
export const ChatMessageContentItemFile$outboundSchema = z.object({
    type: ChatMessageContentItemFileType$outboundSchema,
    file: z.lazy(() => FileT$outboundSchema),
});
export function chatMessageContentItemFileToJSON(chatMessageContentItemFile) {
    return JSON.stringify(ChatMessageContentItemFile$outboundSchema.parse(chatMessageContentItemFile));
}
export function chatMessageContentItemFileFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatMessageContentItemFile$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatMessageContentItemFile' from JSON`);
}
