// @openrouter/sdk/esm/models/chatmessagecontentitemvideo.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 5a81301cb4a7
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { VideoInput$inboundSchema, VideoInput$outboundSchema, } from "./videoinput.js";
/** @internal */
export const ChatMessageContentItemVideo$inboundSchema = z.object({
    type: z.literal("video_url"),
    video_url: VideoInput$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "video_url": "videoUrl",
    });
});
/** @internal */
export const ChatMessageContentItemVideo$outboundSchema = z.object({
    type: z.literal("video_url"),
    videoUrl: VideoInput$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        videoUrl: "video_url",
    });
});
export function chatMessageContentItemVideoToJSON(chatMessageContentItemVideo) {
    return JSON.stringify(ChatMessageContentItemVideo$outboundSchema.parse(chatMessageContentItemVideo));
}
export function chatMessageContentItemVideoFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatMessageContentItemVideo$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatMessageContentItemVideo' from JSON`);
}
