// @openrouter/sdk/esm/models/chatgenerationparams.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: f98a2a558f3f
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import * as openEnums from "../types/enums.js";
import { ChatStreamOptions$outboundSchema, } from "./chatstreamoptions.js";
import { ContextCompressionEngine$outboundSchema, } from "./contextcompressionengine.js";
import { DataCollection$outboundSchema, } from "./datacollection.js";
import { DebugOptions$outboundSchema, } from "./debugoptions.js";
import { Message$outboundSchema, } from "./message.js";
import { PDFParserOptions$outboundSchema, } from "./pdfparseroptions.js";
import { PreferredMaxLatency$outboundSchema, } from "./preferredmaxlatency.js";
import { PreferredMinThroughput$outboundSchema, } from "./preferredminthroughput.js";
import { ProviderName$outboundSchema } from "./providername.js";
import { Quantization$outboundSchema } from "./quantization.js";
import { ResponseFormatJSONObject$outboundSchema, } from "./responseformatjsonobject.js";
import { ResponseFormatJSONSchema$outboundSchema, } from "./responseformatjsonschema.js";
import { ResponseFormatText$outboundSchema, } from "./responseformattext.js";
import { ResponseFormatTextGrammar$outboundSchema, } from "./responseformattextgrammar.js";
import { ResponseFormatTextPython$outboundSchema, } from "./responseformattextpython.js";
import { ToolChoiceOption$outboundSchema, } from "./toolchoiceoption.js";
import { ToolDefinitionJson$outboundSchema, } from "./tooldefinitionjson.js";
import { WebSearchEngine$outboundSchema, } from "./websearchengine.js";
export const ChatGenerationParamsSortEnum = {
    Price: "price",
    Throughput: "throughput",
    Latency: "latency",
    Exacto: "exacto",
};
export const ChatGenerationParamsProviderSortConfigEnum = {
    Price: "price",
    Throughput: "throughput",
    Latency: "latency",
    Exacto: "exacto",
};
/**
 * The provider sorting strategy (price, throughput, latency)
 */
export const ChatGenerationParamsBy = {
    Price: "price",
    Throughput: "throughput",
    Latency: "latency",
    Exacto: "exacto",
};
/**
 * Partitioning strategy for sorting: "model" (default) groups endpoints by model before sorting (fallback models remain fallbacks), "none" sorts all endpoints together regardless of model.
 */
export const ChatGenerationParamsPartition = {
    Model: "model",
    None: "none",
};
/**
 * The provider sorting strategy (price, throughput, latency)
 */
export const ChatGenerationParamsProviderSort = {
    Price: "price",
    Throughput: "throughput",
    Latency: "latency",
    Exacto: "exacto",
};
/**
 * Constrains effort on reasoning for reasoning models
 */
export const Effort = {
    Xhigh: "xhigh",
    High: "high",
    Medium: "medium",
    Low: "low",
    Minimal: "minimal",
    None: "none",
};
export const Modality = {
    Text: "text",
    Image: "image",
    Audio: "audio",
};
export const ChatGenerationParamsType = {
    Ephemeral: "ephemeral",
};
export const ChatGenerationParamsTtl = {
    Fivem: "5m",
    Oneh: "1h",
};
/** @internal */
export const ChatGenerationParamsOrder$outboundSchema = z.union([ProviderName$outboundSchema, z.string()]);
export function chatGenerationParamsOrderToJSON(chatGenerationParamsOrder) {
    return JSON.stringify(ChatGenerationParamsOrder$outboundSchema.parse(chatGenerationParamsOrder));
}
/** @internal */
export const ChatGenerationParamsOnly$outboundSchema = z.union([ProviderName$outboundSchema, z.string()]);
export function chatGenerationParamsOnlyToJSON(chatGenerationParamsOnly) {
    return JSON.stringify(ChatGenerationParamsOnly$outboundSchema.parse(chatGenerationParamsOnly));
}
/** @internal */
export const ChatGenerationParamsIgnore$outboundSchema = z.union([ProviderName$outboundSchema, z.string()]);
export function chatGenerationParamsIgnoreToJSON(chatGenerationParamsIgnore) {
    return JSON.stringify(ChatGenerationParamsIgnore$outboundSchema.parse(chatGenerationParamsIgnore));
}
/** @internal */
export const ChatGenerationParamsSortEnum$outboundSchema = openEnums.outboundSchema(ChatGenerationParamsSortEnum);
/** @internal */
export const ChatGenerationParamsProviderSortConfigEnum$outboundSchema = z.enum(ChatGenerationParamsProviderSortConfigEnum);
/** @internal */
export const ChatGenerationParamsBy$outboundSchema = openEnums.outboundSchema(ChatGenerationParamsBy);
/** @internal */
export const ChatGenerationParamsPartition$outboundSchema = openEnums.outboundSchema(ChatGenerationParamsPartition);
/** @internal */
export const ChatGenerationParamsProviderSortConfig$outboundSchema = z.object({
    by: z.nullable(ChatGenerationParamsBy$outboundSchema).optional(),
    partition: z.nullable(ChatGenerationParamsPartition$outboundSchema)
        .optional(),
});
export function chatGenerationParamsProviderSortConfigToJSON(chatGenerationParamsProviderSortConfig) {
    return JSON.stringify(ChatGenerationParamsProviderSortConfig$outboundSchema.parse(chatGenerationParamsProviderSortConfig));
}
/** @internal */
export const ChatGenerationParamsProviderSortConfigUnion$outboundSchema = z.union([
    z.lazy(() => ChatGenerationParamsProviderSortConfig$outboundSchema),
    ChatGenerationParamsProviderSortConfigEnum$outboundSchema,
]);
export function chatGenerationParamsProviderSortConfigUnionToJSON(chatGenerationParamsProviderSortConfigUnion) {
    return JSON.stringify(ChatGenerationParamsProviderSortConfigUnion$outboundSchema.parse(chatGenerationParamsProviderSortConfigUnion));
}
/** @internal */
export const ChatGenerationParamsProviderSort$outboundSchema = openEnums.outboundSchema(ChatGenerationParamsProviderSort);
/** @internal */
export const ChatGenerationParamsSortUnion$outboundSchema = z.union([
    ChatGenerationParamsProviderSort$outboundSchema,
    z.union([
        z.lazy(() => ChatGenerationParamsProviderSortConfig$outboundSchema),
        ChatGenerationParamsProviderSortConfigEnum$outboundSchema,
    ]),
    ChatGenerationParamsSortEnum$outboundSchema,
]);
export function chatGenerationParamsSortUnionToJSON(chatGenerationParamsSortUnion) {
    return JSON.stringify(ChatGenerationParamsSortUnion$outboundSchema.parse(chatGenerationParamsSortUnion));
}
/** @internal */
export const ChatGenerationParamsMaxPrice$outboundSchema = z.object({
    prompt: z.string().optional(),
    completion: z.string().optional(),
    image: z.string().optional(),
    audio: z.string().optional(),
    request: z.string().optional(),
});
export function chatGenerationParamsMaxPriceToJSON(chatGenerationParamsMaxPrice) {
    return JSON.stringify(ChatGenerationParamsMaxPrice$outboundSchema.parse(chatGenerationParamsMaxPrice));
}
/** @internal */
export const ChatGenerationParamsProvider$outboundSchema = z.object({
    allowFallbacks: z.nullable(z.boolean()).optional(),
    requireParameters: z.nullable(z.boolean()).optional(),
    dataCollection: z.nullable(DataCollection$outboundSchema).optional(),
    zdr: z.nullable(z.boolean()).optional(),
    enforceDistillableText: z.nullable(z.boolean()).optional(),
    order: z.nullable(z.array(z.union([ProviderName$outboundSchema, z.string()])))
        .optional(),
    only: z.nullable(z.array(z.union([ProviderName$outboundSchema, z.string()])))
        .optional(),
    ignore: z.nullable(z.array(z.union([ProviderName$outboundSchema, z.string()]))).optional(),
    quantizations: z.nullable(z.array(Quantization$outboundSchema)).optional(),
    sort: z.nullable(z.union([
        ChatGenerationParamsProviderSort$outboundSchema,
        z.union([
            z.lazy(() => ChatGenerationParamsProviderSortConfig$outboundSchema),
            ChatGenerationParamsProviderSortConfigEnum$outboundSchema,
        ]),
        ChatGenerationParamsSortEnum$outboundSchema,
    ])).optional(),
    maxPrice: z.lazy(() => ChatGenerationParamsMaxPrice$outboundSchema)
        .optional(),
    preferredMinThroughput: z.nullable(PreferredMinThroughput$outboundSchema)
        .optional(),
    preferredMaxLatency: z.nullable(PreferredMaxLatency$outboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        allowFallbacks: "allow_fallbacks",
        requireParameters: "require_parameters",
        dataCollection: "data_collection",
        enforceDistillableText: "enforce_distillable_text",
        maxPrice: "max_price",
        preferredMinThroughput: "preferred_min_throughput",
        preferredMaxLatency: "preferred_max_latency",
    });
});
export function chatGenerationParamsProviderToJSON(chatGenerationParamsProvider) {
    return JSON.stringify(ChatGenerationParamsProvider$outboundSchema.parse(chatGenerationParamsProvider));
}
/** @internal */
export const ChatGenerationParamsPluginContextCompression$outboundSchema = z.object({
    id: z.literal("context-compression"),
    enabled: z.boolean().optional(),
    engine: ContextCompressionEngine$outboundSchema.optional(),
});
export function chatGenerationParamsPluginContextCompressionToJSON(chatGenerationParamsPluginContextCompression) {
    return JSON.stringify(ChatGenerationParamsPluginContextCompression$outboundSchema.parse(chatGenerationParamsPluginContextCompression));
}
/** @internal */
export const ChatGenerationParamsPluginResponseHealing$outboundSchema = z.object({
    id: z.literal("response-healing"),
    enabled: z.boolean().optional(),
});
export function chatGenerationParamsPluginResponseHealingToJSON(chatGenerationParamsPluginResponseHealing) {
    return JSON.stringify(ChatGenerationParamsPluginResponseHealing$outboundSchema.parse(chatGenerationParamsPluginResponseHealing));
}
/** @internal */
export const ChatGenerationParamsPluginFileParser$outboundSchema = z.object({
    id: z.literal("file-parser"),
    enabled: z.boolean().optional(),
    pdf: PDFParserOptions$outboundSchema.optional(),
});
export function chatGenerationParamsPluginFileParserToJSON(chatGenerationParamsPluginFileParser) {
    return JSON.stringify(ChatGenerationParamsPluginFileParser$outboundSchema.parse(chatGenerationParamsPluginFileParser));
}
/** @internal */
export const ChatGenerationParamsPluginWeb$outboundSchema = z.object({
    id: z.literal("web"),
    enabled: z.boolean().optional(),
    maxResults: z.number().optional(),
    searchPrompt: z.string().optional(),
    engine: WebSearchEngine$outboundSchema.optional(),
    includeDomains: z.array(z.string()).optional(),
    excludeDomains: z.array(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        maxResults: "max_results",
        searchPrompt: "search_prompt",
        includeDomains: "include_domains",
        excludeDomains: "exclude_domains",
    });
});
export function chatGenerationParamsPluginWebToJSON(chatGenerationParamsPluginWeb) {
    return JSON.stringify(ChatGenerationParamsPluginWeb$outboundSchema.parse(chatGenerationParamsPluginWeb));
}
/** @internal */
export const ChatGenerationParamsPluginModeration$outboundSchema = z.object({
    id: z.literal("moderation"),
});
export function chatGenerationParamsPluginModerationToJSON(chatGenerationParamsPluginModeration) {
    return JSON.stringify(ChatGenerationParamsPluginModeration$outboundSchema.parse(chatGenerationParamsPluginModeration));
}
/** @internal */
export const ChatGenerationParamsPluginAutoRouter$outboundSchema = z.object({
    id: z.literal("auto-router"),
    enabled: z.boolean().optional(),
    allowedModels: z.array(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        allowedModels: "allowed_models",
    });
});
export function chatGenerationParamsPluginAutoRouterToJSON(chatGenerationParamsPluginAutoRouter) {
    return JSON.stringify(ChatGenerationParamsPluginAutoRouter$outboundSchema.parse(chatGenerationParamsPluginAutoRouter));
}
/** @internal */
export const ChatGenerationParamsPluginUnion$outboundSchema = z.union([
    z.lazy(() => ChatGenerationParamsPluginAutoRouter$outboundSchema),
    z.lazy(() => ChatGenerationParamsPluginModeration$outboundSchema),
    z.lazy(() => ChatGenerationParamsPluginWeb$outboundSchema),
    z.lazy(() => ChatGenerationParamsPluginFileParser$outboundSchema),
    z.lazy(() => ChatGenerationParamsPluginResponseHealing$outboundSchema),
    z.lazy(() => ChatGenerationParamsPluginContextCompression$outboundSchema),
]);
export function chatGenerationParamsPluginUnionToJSON(chatGenerationParamsPluginUnion) {
    return JSON.stringify(ChatGenerationParamsPluginUnion$outboundSchema.parse(chatGenerationParamsPluginUnion));
}
/** @internal */
export const ChatGenerationParamsTrace$outboundSchema = z.object({
    traceId: z.string().optional(),
    traceName: z.string().optional(),
    spanName: z.string().optional(),
    generationName: z.string().optional(),
    parentSpanId: z.string().optional(),
    additionalProperties: z.record(z.string(), z.nullable(z.any())).optional(),
}).transform((v) => {
    return {
        ...v.additionalProperties,
        ...remap$(v, {
            traceId: "trace_id",
            traceName: "trace_name",
            spanName: "span_name",
            generationName: "generation_name",
            parentSpanId: "parent_span_id",
            additionalProperties: null,
        }),
    };
});
export function chatGenerationParamsTraceToJSON(chatGenerationParamsTrace) {
    return JSON.stringify(ChatGenerationParamsTrace$outboundSchema.parse(chatGenerationParamsTrace));
}
/** @internal */
export const Effort$outboundSchema = openEnums
    .outboundSchema(Effort);
/** @internal */
export const Reasoning$outboundSchema = z.object({
    effort: z.nullable(Effort$outboundSchema).optional(),
    summary: z.nullable(z.any()).optional(),
});
export function reasoningToJSON(reasoning) {
    return JSON.stringify(Reasoning$outboundSchema.parse(reasoning));
}
/** @internal */
export const ResponseFormat$outboundSchema = z.union([
    ResponseFormatText$outboundSchema,
    ResponseFormatJSONObject$outboundSchema,
    ResponseFormatJSONSchema$outboundSchema,
    ResponseFormatTextGrammar$outboundSchema,
    ResponseFormatTextPython$outboundSchema,
]);
export function responseFormatToJSON(responseFormat) {
    return JSON.stringify(ResponseFormat$outboundSchema.parse(responseFormat));
}
/** @internal */
export const Stop$outboundSchema = z.union([
    z.string(),
    z.array(z.string()),
    z.any(),
]);
export function stopToJSON(stop) {
    return JSON.stringify(Stop$outboundSchema.parse(stop));
}
/** @internal */
export const ChatGenerationParamsImageConfig$outboundSchema = z.union([z.string(), z.number(), z.array(z.nullable(z.any()))]);
export function chatGenerationParamsImageConfigToJSON(chatGenerationParamsImageConfig) {
    return JSON.stringify(ChatGenerationParamsImageConfig$outboundSchema.parse(chatGenerationParamsImageConfig));
}
/** @internal */
export const Modality$outboundSchema = openEnums
    .outboundSchema(Modality);
/** @internal */
export const ChatGenerationParamsType$outboundSchema = z.enum(ChatGenerationParamsType);
/** @internal */
export const ChatGenerationParamsTtl$outboundSchema = openEnums.outboundSchema(ChatGenerationParamsTtl);
/** @internal */
export const CacheControl$outboundSchema = z.object({
    type: ChatGenerationParamsType$outboundSchema,
    ttl: ChatGenerationParamsTtl$outboundSchema.optional(),
});
export function cacheControlToJSON(cacheControl) {
    return JSON.stringify(CacheControl$outboundSchema.parse(cacheControl));
}
/** @internal */
export const ChatGenerationParams$outboundSchema = z.object({
    provider: z.nullable(z.lazy(() => ChatGenerationParamsProvider$outboundSchema)).optional(),
    plugins: z.array(z.union([
        z.lazy(() => ChatGenerationParamsPluginAutoRouter$outboundSchema),
        z.lazy(() => ChatGenerationParamsPluginModeration$outboundSchema),
        z.lazy(() => ChatGenerationParamsPluginWeb$outboundSchema),
        z.lazy(() => ChatGenerationParamsPluginFileParser$outboundSchema),
        z.lazy(() => ChatGenerationParamsPluginResponseHealing$outboundSchema),
        z.lazy(() => ChatGenerationParamsPluginContextCompression$outboundSchema),
    ])).optional(),
    user: z.string().optional(),
    sessionId: z.string().optional(),
    trace: z.lazy(() => ChatGenerationParamsTrace$outboundSchema).optional(),
    messages: z.array(Message$outboundSchema),
    model: z.string().optional(),
    models: z.array(z.string()).optional(),
    frequencyPenalty: z.nullable(z.number()).optional(),
    logitBias: z.nullable(z.record(z.string(), z.number())).optional(),
    logprobs: z.nullable(z.boolean()).optional(),
    topLogprobs: z.nullable(z.number()).optional(),
    maxCompletionTokens: z.nullable(z.number()).optional(),
    maxTokens: z.nullable(z.number()).optional(),
    metadata: z.record(z.string(), z.string()).optional(),
    presencePenalty: z.nullable(z.number()).optional(),
    reasoning: z.lazy(() => Reasoning$outboundSchema).optional(),
    responseFormat: z.union([
        ResponseFormatText$outboundSchema,
        ResponseFormatJSONObject$outboundSchema,
        ResponseFormatJSONSchema$outboundSchema,
        ResponseFormatTextGrammar$outboundSchema,
        ResponseFormatTextPython$outboundSchema,
    ]).optional(),
    seed: z.nullable(z.int()).optional(),
    stop: z.nullable(z.union([z.string(), z.array(z.string()), z.any()]))
        .optional(),
    stream: z.boolean().default(false),
    streamOptions: z.nullable(ChatStreamOptions$outboundSchema).optional(),
    temperature: z.nullable(z.number().default(1)),
    parallelToolCalls: z.nullable(z.boolean()).optional(),
    toolChoice: ToolChoiceOption$outboundSchema.optional(),
    tools: z.array(ToolDefinitionJson$outboundSchema).optional(),
    topP: z.nullable(z.number().default(1)),
    debug: DebugOptions$outboundSchema.optional(),
    imageConfig: z.record(z.string(), z.union([z.string(), z.number(), z.array(z.nullable(z.any()))])).optional(),
    modalities: z.array(Modality$outboundSchema).optional(),
    cacheControl: z.lazy(() => CacheControl$outboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        sessionId: "session_id",
        frequencyPenalty: "frequency_penalty",
        logitBias: "logit_bias",
        topLogprobs: "top_logprobs",
        maxCompletionTokens: "max_completion_tokens",
        maxTokens: "max_tokens",
        presencePenalty: "presence_penalty",
        responseFormat: "response_format",
        streamOptions: "stream_options",
        parallelToolCalls: "parallel_tool_calls",
        toolChoice: "tool_choice",
        topP: "top_p",
        imageConfig: "image_config",
        cacheControl: "cache_control",
    });
});
export function chatGenerationParamsToJSON(chatGenerationParams) {
    return JSON.stringify(ChatGenerationParams$outboundSchema.parse(chatGenerationParams));
}
