// @openrouter/sdk/esm/models/chatmessagecontentitemaudio.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 49b1cafcb338
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
export const ChatMessageContentItemAudioType = {
    InputAudio: "input_audio",
};
/** @internal */
export const ChatMessageContentItemAudioType$inboundSchema = z.enum(ChatMessageContentItemAudioType);
/** @internal */
export const ChatMessageContentItemAudioType$outboundSchema = ChatMessageContentItemAudioType$inboundSchema;
/** @internal */
export const ChatMessageContentItemAudioInputAudio$inboundSchema = z.object({
    data: z.string(),
    format: z.string(),
});
/** @internal */
export const ChatMessageContentItemAudioInputAudio$outboundSchema = z.object({
    data: z.string(),
    format: z.string(),
});
export function chatMessageContentItemAudioInputAudioToJSON(chatMessageContentItemAudioInputAudio) {
    return JSON.stringify(ChatMessageContentItemAudioInputAudio$outboundSchema.parse(chatMessageContentItemAudioInputAudio));
}
export function chatMessageContentItemAudioInputAudioFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatMessageContentItemAudioInputAudio$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatMessageContentItemAudioInputAudio' from JSON`);
}
/** @internal */
export const ChatMessageContentItemAudio$inboundSchema = z.object({
    type: ChatMessageContentItemAudioType$inboundSchema,
    input_audio: z.lazy(() => ChatMessageContentItemAudioInputAudio$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "input_audio": "inputAudio",
    });
});
/** @internal */
export const ChatMessageContentItemAudio$outboundSchema = z.object({
    type: ChatMessageContentItemAudioType$outboundSchema,
    inputAudio: z.lazy(() => ChatMessageContentItemAudioInputAudio$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        inputAudio: "input_audio",
    });
});
export function chatMessageContentItemAudioToJSON(chatMessageContentItemAudio) {
    return JSON.stringify(ChatMessageContentItemAudio$outboundSchema.parse(chatMessageContentItemAudio));
}
export function chatMessageContentItemAudioFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatMessageContentItemAudio$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatMessageContentItemAudio' from JSON`);
}
