// @openrouter/sdk/esm/models/chatmessagecontentitemcachecontrol.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: b5c18e04e19c
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
export const ChatMessageContentItemCacheControlType = {
    Ephemeral: "ephemeral",
};
export const ChatMessageContentItemCacheControlTtl = {
    Fivem: "5m",
    Oneh: "1h",
};
/** @internal */
export const ChatMessageContentItemCacheControlType$inboundSchema = z.enum(ChatMessageContentItemCacheControlType);
/** @internal */
export const ChatMessageContentItemCacheControlType$outboundSchema = ChatMessageContentItemCacheControlType$inboundSchema;
/** @internal */
export const ChatMessageContentItemCacheControlTtl$inboundSchema = openEnums.inboundSchema(ChatMessageContentItemCacheControlTtl);
/** @internal */
export const ChatMessageContentItemCacheControlTtl$outboundSchema = openEnums.outboundSchema(ChatMessageContentItemCacheControlTtl);
/** @internal */
export const ChatMessageContentItemCacheControl$inboundSchema = z.object({
    type: ChatMessageContentItemCacheControlType$inboundSchema,
    ttl: ChatMessageContentItemCacheControlTtl$inboundSchema.optional(),
});
/** @internal */
export const ChatMessageContentItemCacheControl$outboundSchema = z.object({
    type: ChatMessageContentItemCacheControlType$outboundSchema,
    ttl: ChatMessageContentItemCacheControlTtl$outboundSchema.optional(),
});
export function chatMessageContentItemCacheControlToJSON(chatMessageContentItemCacheControl) {
    return JSON.stringify(ChatMessageContentItemCacheControl$outboundSchema.parse(chatMessageContentItemCacheControl));
}
export function chatMessageContentItemCacheControlFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatMessageContentItemCacheControl$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatMessageContentItemCacheControl' from JSON`);
}
