// @openrouter/sdk/esm/models/chatcompletionaudiooutput.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 125d1fe30400
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const ChatCompletionAudioOutput$inboundSchema = z.object({
    id: z.string().optional(),
    expires_at: z.number().optional(),
    data: z.string().optional(),
    transcript: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        "expires_at": "expiresAt",
    });
});
/** @internal */
export const ChatCompletionAudioOutput$outboundSchema = z.object({
    id: z.string().optional(),
    expiresAt: z.number().optional(),
    data: z.string().optional(),
    transcript: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        expiresAt: "expires_at",
    });
});
export function chatCompletionAudioOutputToJSON(chatCompletionAudioOutput) {
    return JSON.stringify(ChatCompletionAudioOutput$outboundSchema.parse(chatCompletionAudioOutput));
}
export function chatCompletionAudioOutputFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatCompletionAudioOutput$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatCompletionAudioOutput' from JSON`);
}
