// @openrouter/sdk/esm/models/chatmessagecontentitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: bb7f6b29b93f
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import { ChatMessageContentItemAudio$inboundSchema, ChatMessageContentItemAudio$outboundSchema, } from "./chatmessagecontentitemaudio.js";
import { ChatMessageContentItemFile$inboundSchema, ChatMessageContentItemFile$outboundSchema, } from "./chatmessagecontentitemfile.js";
import { ChatMessageContentItemImage$inboundSchema, ChatMessageContentItemImage$outboundSchema, } from "./chatmessagecontentitemimage.js";
import { ChatMessageContentItemText$inboundSchema, ChatMessageContentItemText$outboundSchema, } from "./chatmessagecontentitemtext.js";
import { ChatMessageContentItemVideo$inboundSchema, ChatMessageContentItemVideo$outboundSchema, } from "./chatmessagecontentitemvideo.js";
import { ChatMessageContentItemVideoLegacy$inboundSchema, ChatMessageContentItemVideoLegacy$outboundSchema, } from "./chatmessagecontentitemvideolegacy.js";
/** @internal */
export const ChatMessageContentItem1$inboundSchema = z.union([
    ChatMessageContentItemVideoLegacy$inboundSchema,
    ChatMessageContentItemVideo$inboundSchema,
]);
/** @internal */
export const ChatMessageContentItem1$outboundSchema = z.union([
    ChatMessageContentItemVideoLegacy$outboundSchema,
    ChatMessageContentItemVideo$outboundSchema,
]);
export function chatMessageContentItem1ToJSON(chatMessageContentItem1) {
    return JSON.stringify(ChatMessageContentItem1$outboundSchema.parse(chatMessageContentItem1));
}
export function chatMessageContentItem1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatMessageContentItem1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatMessageContentItem1' from JSON`);
}
/** @internal */
export const ChatMessageContentItem$inboundSchema = z.union([
    ChatMessageContentItemText$inboundSchema,
    ChatMessageContentItemImage$inboundSchema,
    ChatMessageContentItemAudio$inboundSchema,
    ChatMessageContentItemFile$inboundSchema,
    z.union([
        ChatMessageContentItemVideoLegacy$inboundSchema,
        ChatMessageContentItemVideo$inboundSchema,
    ]),
]);
/** @internal */
export const ChatMessageContentItem$outboundSchema = z.union([
    ChatMessageContentItemText$outboundSchema,
    ChatMessageContentItemImage$outboundSchema,
    ChatMessageContentItemAudio$outboundSchema,
    ChatMessageContentItemFile$outboundSchema,
    z.union([
        ChatMessageContentItemVideoLegacy$outboundSchema,
        ChatMessageContentItemVideo$outboundSchema,
    ]),
]);
export function chatMessageContentItemToJSON(chatMessageContentItem) {
    return JSON.stringify(ChatMessageContentItem$outboundSchema.parse(chatMessageContentItem));
}
export function chatMessageContentItemFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatMessageContentItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatMessageContentItem' from JSON`);
}
