// @openrouter/sdk/esm/models/chatstreamingresponsechunk.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 00bf2ea52439
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ChatGenerationTokenUsage$inboundSchema, } from "./chatgenerationtokenusage.js";
import { ChatStreamingChoice$inboundSchema, } from "./chatstreamingchoice.js";
export const ChatStreamingResponseChunkObject = {
    ChatCompletionChunk: "chat.completion.chunk",
};
/** @internal */
export const ChatStreamingResponseChunkObject$inboundSchema = z.enum(ChatStreamingResponseChunkObject);
/** @internal */
export const ErrorT$inboundSchema = z.object({
    message: z.string(),
    code: z.number(),
});
export function errorFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ErrorT$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ErrorT' from JSON`);
}
/** @internal */
export const ChatStreamingResponseChunk$inboundSchema = z.object({
    id: z.string(),
    choices: z.array(ChatStreamingChoice$inboundSchema),
    created: z.number(),
    model: z.string(),
    object: ChatStreamingResponseChunkObject$inboundSchema,
    system_fingerprint: z.string().optional(),
    error: z.lazy(() => ErrorT$inboundSchema).optional(),
    usage: ChatGenerationTokenUsage$inboundSchema.optional(),
}).transform((v) => {
    return remap$(v, {
        "system_fingerprint": "systemFingerprint",
    });
});
export function chatStreamingResponseChunkFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatStreamingResponseChunk$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatStreamingResponseChunk' from JSON`);
}
