// @openrouter/sdk/esm/models/chatresponse.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 76c3703fa3d5
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ChatGenerationTokenUsage$inboundSchema, } from "./chatgenerationtokenusage.js";
import { ChatResponseChoice$inboundSchema, } from "./chatresponsechoice.js";
export const ChatResponseObject = {
    ChatCompletion: "chat.completion",
};
/** @internal */
export const ChatResponseObject$inboundSchema = z.enum(ChatResponseObject);
/** @internal */
export const ChatResponse$inboundSchema = z
    .object({
    id: z.string(),
    choices: z.array(ChatResponseChoice$inboundSchema),
    created: z.number(),
    model: z.string(),
    object: ChatResponseObject$inboundSchema,
    system_fingerprint: z.nullable(z.string()),
    usage: ChatGenerationTokenUsage$inboundSchema.optional(),
}).transform((v) => {
    return remap$(v, {
        "system_fingerprint": "systemFingerprint",
    });
});
export function chatResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatResponse' from JSON`);
}
