// @openrouter/sdk/esm/models/openresponsesfunctioncalloutput.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 5e801f7b2903
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import * as openEnums from "../types/enums.js";
import { ResponseInputFile$outboundSchema, } from "./responseinputfile.js";
import { ResponseInputText$outboundSchema, } from "./responseinputtext.js";
import { ToolCallStatus$outboundSchema, } from "./toolcallstatus.js";
export const OpenResponsesFunctionCallOutputTypeFunctionCallOutput = {
    FunctionCallOutput: "function_call_output",
};
export const OpenResponsesFunctionCallOutputDetail = {
    Auto: "auto",
    High: "high",
    Low: "low",
};
/** @internal */
export const OpenResponsesFunctionCallOutputTypeFunctionCallOutput$outboundSchema = z
    .enum(OpenResponsesFunctionCallOutputTypeFunctionCallOutput);
/** @internal */
export const OpenResponsesFunctionCallOutputDetail$outboundSchema = openEnums.outboundSchema(OpenResponsesFunctionCallOutputDetail);
/** @internal */
export const OutputInputImage$outboundSchema = z.object({
    type: z.literal("input_image"),
    detail: OpenResponsesFunctionCallOutputDetail$outboundSchema,
    imageUrl: z.nullable(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        imageUrl: "image_url",
    });
});
export function outputInputImageToJSON(outputInputImage) {
    return JSON.stringify(OutputInputImage$outboundSchema.parse(outputInputImage));
}
/** @internal */
export const OpenResponsesFunctionCallOutputOutputUnion1$outboundSchema = z.union([
    ResponseInputText$outboundSchema,
    z.lazy(() => OutputInputImage$outboundSchema),
    ResponseInputFile$outboundSchema,
]);
export function openResponsesFunctionCallOutputOutputUnion1ToJSON(openResponsesFunctionCallOutputOutputUnion1) {
    return JSON.stringify(OpenResponsesFunctionCallOutputOutputUnion1$outboundSchema.parse(openResponsesFunctionCallOutputOutputUnion1));
}
/** @internal */
export const OpenResponsesFunctionCallOutputOutputUnion2$outboundSchema = z.union([
    z.string(),
    z.array(z.union([
        ResponseInputText$outboundSchema,
        z.lazy(() => OutputInputImage$outboundSchema),
        ResponseInputFile$outboundSchema,
    ])),
]);
export function openResponsesFunctionCallOutputOutputUnion2ToJSON(openResponsesFunctionCallOutputOutputUnion2) {
    return JSON.stringify(OpenResponsesFunctionCallOutputOutputUnion2$outboundSchema.parse(openResponsesFunctionCallOutputOutputUnion2));
}
/** @internal */
export const OpenResponsesFunctionCallOutput$outboundSchema = z.object({
    type: OpenResponsesFunctionCallOutputTypeFunctionCallOutput$outboundSchema,
    id: z.nullable(z.string()).optional(),
    callId: z.string(),
    output: z.union([
        z.string(),
        z.array(z.union([
            ResponseInputText$outboundSchema,
            z.lazy(() => OutputInputImage$outboundSchema),
            ResponseInputFile$outboundSchema,
        ])),
    ]),
    status: z.nullable(ToolCallStatus$outboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        callId: "call_id",
    });
});
export function openResponsesFunctionCallOutputToJSON(openResponsesFunctionCallOutput) {
    return JSON.stringify(OpenResponsesFunctionCallOutput$outboundSchema.parse(openResponsesFunctionCallOutput));
}
