// @openrouter/sdk/esm/models/openresponsesmcptool.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 1693a27fc478
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
export const ConnectorId = {
    ConnectorDropbox: "connector_dropbox",
    ConnectorGmail: "connector_gmail",
    ConnectorGooglecalendar: "connector_googlecalendar",
    ConnectorGoogledrive: "connector_googledrive",
    ConnectorMicrosoftteams: "connector_microsoftteams",
    ConnectorOutlookcalendar: "connector_outlookcalendar",
    ConnectorOutlookemail: "connector_outlookemail",
    ConnectorSharepoint: "connector_sharepoint",
};
export const RequireApprovalNever = {
    Never: "never",
};
export const RequireApprovalAlways = {
    Always: "always",
};
/** @internal */
export const AllowedTools$inboundSchema = z
    .object({
    tool_names: z.array(z.string()).optional(),
    read_only: z.boolean().optional(),
}).transform((v) => {
    return remap$(v, {
        "tool_names": "toolNames",
        "read_only": "readOnly",
    });
});
/** @internal */
export const AllowedTools$outboundSchema = z.object({
    toolNames: z.array(z.string()).optional(),
    readOnly: z.boolean().optional(),
}).transform((v) => {
    return remap$(v, {
        toolNames: "tool_names",
        readOnly: "read_only",
    });
});
export function allowedToolsToJSON(allowedTools) {
    return JSON.stringify(AllowedTools$outboundSchema.parse(allowedTools));
}
export function allowedToolsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => AllowedTools$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AllowedTools' from JSON`);
}
/** @internal */
export const ConnectorId$inboundSchema = openEnums.inboundSchema(ConnectorId);
/** @internal */
export const ConnectorId$outboundSchema = openEnums.outboundSchema(ConnectorId);
/** @internal */
export const RequireApprovalNever$inboundSchema = z.enum(RequireApprovalNever);
/** @internal */
export const RequireApprovalNever$outboundSchema = RequireApprovalNever$inboundSchema;
/** @internal */
export const RequireApprovalAlways$inboundSchema = z.enum(RequireApprovalAlways);
/** @internal */
export const RequireApprovalAlways$outboundSchema = RequireApprovalAlways$inboundSchema;
/** @internal */
export const Never$inboundSchema = z.object({
    tool_names: z.array(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        "tool_names": "toolNames",
    });
});
/** @internal */
export const Never$outboundSchema = z.object({
    toolNames: z.array(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        toolNames: "tool_names",
    });
});
export function neverToJSON(never) {
    return JSON.stringify(Never$outboundSchema.parse(never));
}
export function neverFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Never$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Never' from JSON`);
}
/** @internal */
export const Always$inboundSchema = z.object({
    tool_names: z.array(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        "tool_names": "toolNames",
    });
});
/** @internal */
export const Always$outboundSchema = z
    .object({
    toolNames: z.array(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        toolNames: "tool_names",
    });
});
export function alwaysToJSON(always) {
    return JSON.stringify(Always$outboundSchema.parse(always));
}
export function alwaysFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Always$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Always' from JSON`);
}
/** @internal */
export const RequireApproval$inboundSchema = z.object({
    never: z.lazy(() => Never$inboundSchema).optional(),
    always: z.lazy(() => Always$inboundSchema).optional(),
});
/** @internal */
export const RequireApproval$outboundSchema = z.object({
    never: z.lazy(() => Never$outboundSchema).optional(),
    always: z.lazy(() => Always$outboundSchema).optional(),
});
export function requireApprovalToJSON(requireApproval) {
    return JSON.stringify(RequireApproval$outboundSchema.parse(requireApproval));
}
export function requireApprovalFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RequireApproval$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RequireApproval' from JSON`);
}
/** @internal */
export const OpenResponsesMcpTool$inboundSchema = z.object({
    type: z.literal("mcp"),
    server_label: z.string(),
    allowed_tools: z.nullable(z.any()).optional(),
    authorization: z.string().optional(),
    connector_id: ConnectorId$inboundSchema.optional(),
    headers: z.nullable(z.record(z.string(), z.string())).optional(),
    require_approval: z.nullable(z.any()).optional(),
    server_description: z.string().optional(),
    server_url: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        "server_label": "serverLabel",
        "allowed_tools": "allowedTools",
        "connector_id": "connectorId",
        "require_approval": "requireApproval",
        "server_description": "serverDescription",
        "server_url": "serverUrl",
    });
});
/** @internal */
export const OpenResponsesMcpTool$outboundSchema = z.object({
    type: z.literal("mcp"),
    serverLabel: z.string(),
    allowedTools: z.nullable(z.any()).optional(),
    authorization: z.string().optional(),
    connectorId: ConnectorId$outboundSchema.optional(),
    headers: z.nullable(z.record(z.string(), z.string())).optional(),
    requireApproval: z.nullable(z.any()).optional(),
    serverDescription: z.string().optional(),
    serverUrl: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        serverLabel: "server_label",
        allowedTools: "allowed_tools",
        connectorId: "connector_id",
        requireApproval: "require_approval",
        serverDescription: "server_description",
        serverUrl: "server_url",
    });
});
export function openResponsesMcpToolToJSON(openResponsesMcpTool) {
    return JSON.stringify(OpenResponsesMcpTool$outboundSchema.parse(openResponsesMcpTool));
}
export function openResponsesMcpToolFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesMcpTool$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesMcpTool' from JSON`);
}
