// @openrouter/sdk/esm/models/openresponseswebsearchpreview20250311tool.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: d450efb8b660
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
import { ResponsesSearchContextSize$inboundSchema, ResponsesSearchContextSize$outboundSchema, } from "./responsessearchcontextsize.js";
import { WebSearchPreviewToolUserLocation$inboundSchema, WebSearchPreviewToolUserLocation$outboundSchema, } from "./websearchpreviewtooluserlocation.js";
/**
 * Which search engine to use. "auto" (default) uses native if the provider supports it, otherwise Exa. "native" forces the provider's built-in search. "exa" forces the Exa search API.
 */
export const OpenResponsesWebSearchPreview20250311ToolEngine = {
    Auto: "auto",
    Native: "native",
    Exa: "exa",
};
/** @internal */
export const OpenResponsesWebSearchPreview20250311ToolEngine$inboundSchema = openEnums.inboundSchema(OpenResponsesWebSearchPreview20250311ToolEngine);
/** @internal */
export const OpenResponsesWebSearchPreview20250311ToolEngine$outboundSchema = openEnums
    .outboundSchema(OpenResponsesWebSearchPreview20250311ToolEngine);
/** @internal */
export const OpenResponsesWebSearchPreview20250311ToolFilters$inboundSchema = z
    .object({
    allowed_domains: z.nullable(z.array(z.string())).optional(),
    excluded_domains: z.nullable(z.array(z.string())).optional(),
}).transform((v) => {
    return remap$(v, {
        "allowed_domains": "allowedDomains",
        "excluded_domains": "excludedDomains",
    });
});
/** @internal */
export const OpenResponsesWebSearchPreview20250311ToolFilters$outboundSchema = z.object({
    allowedDomains: z.nullable(z.array(z.string())).optional(),
    excludedDomains: z.nullable(z.array(z.string())).optional(),
}).transform((v) => {
    return remap$(v, {
        allowedDomains: "allowed_domains",
        excludedDomains: "excluded_domains",
    });
});
export function openResponsesWebSearchPreview20250311ToolFiltersToJSON(openResponsesWebSearchPreview20250311ToolFilters) {
    return JSON.stringify(OpenResponsesWebSearchPreview20250311ToolFilters$outboundSchema.parse(openResponsesWebSearchPreview20250311ToolFilters));
}
export function openResponsesWebSearchPreview20250311ToolFiltersFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesWebSearchPreview20250311ToolFilters$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesWebSearchPreview20250311ToolFilters' from JSON`);
}
/** @internal */
export const OpenResponsesWebSearchPreview20250311Tool$inboundSchema = z.object({
    type: z.literal("web_search_preview_2025_03_11"),
    search_context_size: ResponsesSearchContextSize$inboundSchema.optional(),
    user_location: z.nullable(WebSearchPreviewToolUserLocation$inboundSchema)
        .optional(),
    engine: OpenResponsesWebSearchPreview20250311ToolEngine$inboundSchema
        .optional(),
    max_results: z.number().optional(),
    filters: z.nullable(z.lazy(() => OpenResponsesWebSearchPreview20250311ToolFilters$inboundSchema)).optional(),
}).transform((v) => {
    return remap$(v, {
        "search_context_size": "searchContextSize",
        "user_location": "userLocation",
        "max_results": "maxResults",
    });
});
/** @internal */
export const OpenResponsesWebSearchPreview20250311Tool$outboundSchema = z.object({
    type: z.literal("web_search_preview_2025_03_11"),
    searchContextSize: ResponsesSearchContextSize$outboundSchema.optional(),
    userLocation: z.nullable(WebSearchPreviewToolUserLocation$outboundSchema)
        .optional(),
    engine: OpenResponsesWebSearchPreview20250311ToolEngine$outboundSchema
        .optional(),
    maxResults: z.number().optional(),
    filters: z.nullable(z.lazy(() => OpenResponsesWebSearchPreview20250311ToolFilters$outboundSchema)).optional(),
}).transform((v) => {
    return remap$(v, {
        searchContextSize: "search_context_size",
        userLocation: "user_location",
        maxResults: "max_results",
    });
});
export function openResponsesWebSearchPreview20250311ToolToJSON(openResponsesWebSearchPreview20250311Tool) {
    return JSON.stringify(OpenResponsesWebSearchPreview20250311Tool$outboundSchema.parse(openResponsesWebSearchPreview20250311Tool));
}
export function openResponsesWebSearchPreview20250311ToolFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesWebSearchPreview20250311Tool$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesWebSearchPreview20250311Tool' from JSON`);
}
