// @openrouter/sdk/esm/models/openresponsescustomtool.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 9f957484e6a0
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
export const Syntax = {
    Lark: "lark",
    Regex: "regex",
};
/** @internal */
export const Syntax$inboundSchema = openEnums
    .inboundSchema(Syntax);
/** @internal */
export const Syntax$outboundSchema = openEnums
    .outboundSchema(Syntax);
/** @internal */
export const FormatGrammar$inboundSchema = z
    .object({
    type: z.literal("grammar"),
    definition: z.string(),
    syntax: Syntax$inboundSchema,
});
/** @internal */
export const FormatGrammar$outboundSchema = z.object({
    type: z.literal("grammar"),
    definition: z.string(),
    syntax: Syntax$outboundSchema,
});
export function formatGrammarToJSON(formatGrammar) {
    return JSON.stringify(FormatGrammar$outboundSchema.parse(formatGrammar));
}
export function formatGrammarFromJSON(jsonString) {
    return safeParse(jsonString, (x) => FormatGrammar$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FormatGrammar' from JSON`);
}
/** @internal */
export const FormatText$inboundSchema = z
    .object({
    type: z.literal("text"),
});
/** @internal */
export const FormatText$outboundSchema = z.object({
    type: z.literal("text"),
});
export function formatTextToJSON(formatText) {
    return JSON.stringify(FormatText$outboundSchema.parse(formatText));
}
export function formatTextFromJSON(jsonString) {
    return safeParse(jsonString, (x) => FormatText$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FormatText' from JSON`);
}
/** @internal */
export const Format$inboundSchema = z.union([
    z.lazy(() => FormatText$inboundSchema),
    z.lazy(() => FormatGrammar$inboundSchema),
]);
/** @internal */
export const Format$outboundSchema = z
    .union([
    z.lazy(() => FormatText$outboundSchema),
    z.lazy(() => FormatGrammar$outboundSchema),
]);
export function formatToJSON(format) {
    return JSON.stringify(Format$outboundSchema.parse(format));
}
export function formatFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Format$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Format' from JSON`);
}
/** @internal */
export const OpenResponsesCustomTool$inboundSchema = z.object({
    type: z.literal("custom"),
    name: z.string(),
    description: z.string().optional(),
    format: z.union([
        z.lazy(() => FormatText$inboundSchema),
        z.lazy(() => FormatGrammar$inboundSchema),
    ]).optional(),
});
/** @internal */
export const OpenResponsesCustomTool$outboundSchema = z.object({
    type: z.literal("custom"),
    name: z.string(),
    description: z.string().optional(),
    format: z.union([
        z.lazy(() => FormatText$outboundSchema),
        z.lazy(() => FormatGrammar$outboundSchema),
    ]).optional(),
});
export function openResponsesCustomToolToJSON(openResponsesCustomTool) {
    return JSON.stringify(OpenResponsesCustomTool$outboundSchema.parse(openResponsesCustomTool));
}
export function openResponsesCustomToolFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesCustomTool$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesCustomTool' from JSON`);
}
