// @openrouter/sdk/esm/models/openresponsesfilesearchtool.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 0f461c2f2470
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
import { CompoundFilter$inboundSchema, CompoundFilter$outboundSchema, } from "./compoundfilter.js";
export const FiltersType = {
    Eq: "eq",
    Ne: "ne",
    Gt: "gt",
    Gte: "gte",
    Lt: "lt",
    Lte: "lte",
};
export const Ranker = {
    Auto: "auto",
    Default20241115: "default-2024-11-15",
};
/** @internal */
export const FiltersType$inboundSchema = openEnums.inboundSchema(FiltersType);
/** @internal */
export const FiltersType$outboundSchema = openEnums.outboundSchema(FiltersType);
/** @internal */
export const Value1$inboundSchema = z.union([
    z.string(),
    z.number(),
]);
/** @internal */
export const Value1$outboundSchema = z
    .union([z.string(), z.number()]);
export function value1ToJSON(value1) {
    return JSON.stringify(Value1$outboundSchema.parse(value1));
}
export function value1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Value1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Value1' from JSON`);
}
/** @internal */
export const Value2$inboundSchema = z.union([
    z.string(),
    z.number(),
    z.boolean(),
    z.array(z.union([z.string(), z.number()])),
]);
/** @internal */
export const Value2$outboundSchema = z
    .union([
    z.string(),
    z.number(),
    z.boolean(),
    z.array(z.union([z.string(), z.number()])),
]);
export function value2ToJSON(value2) {
    return JSON.stringify(Value2$outboundSchema.parse(value2));
}
export function value2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Value2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Value2' from JSON`);
}
/** @internal */
export const OpenResponsesFileSearchToolFilters$inboundSchema = z.object({
    key: z.string(),
    type: FiltersType$inboundSchema,
    value: z.union([
        z.string(),
        z.number(),
        z.boolean(),
        z.array(z.union([z.string(), z.number()])),
    ]),
});
/** @internal */
export const OpenResponsesFileSearchToolFilters$outboundSchema = z.object({
    key: z.string(),
    type: FiltersType$outboundSchema,
    value: z.union([
        z.string(),
        z.number(),
        z.boolean(),
        z.array(z.union([z.string(), z.number()])),
    ]),
});
export function openResponsesFileSearchToolFiltersToJSON(openResponsesFileSearchToolFilters) {
    return JSON.stringify(OpenResponsesFileSearchToolFilters$outboundSchema.parse(openResponsesFileSearchToolFilters));
}
export function openResponsesFileSearchToolFiltersFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesFileSearchToolFilters$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesFileSearchToolFilters' from JSON`);
}
/** @internal */
export const Filters$inboundSchema = z.union([
    z.lazy(() => OpenResponsesFileSearchToolFilters$inboundSchema),
    CompoundFilter$inboundSchema,
    z.any(),
]);
/** @internal */
export const Filters$outboundSchema = z
    .union([
    z.lazy(() => OpenResponsesFileSearchToolFilters$outboundSchema),
    CompoundFilter$outboundSchema,
    z.any(),
]);
export function filtersToJSON(filters) {
    return JSON.stringify(Filters$outboundSchema.parse(filters));
}
export function filtersFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Filters$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Filters' from JSON`);
}
/** @internal */
export const Ranker$inboundSchema = openEnums
    .inboundSchema(Ranker);
/** @internal */
export const Ranker$outboundSchema = openEnums
    .outboundSchema(Ranker);
/** @internal */
export const RankingOptions$inboundSchema = z.object({
    ranker: Ranker$inboundSchema.optional(),
    score_threshold: z.number().optional(),
}).transform((v) => {
    return remap$(v, {
        "score_threshold": "scoreThreshold",
    });
});
/** @internal */
export const RankingOptions$outboundSchema = z.object({
    ranker: Ranker$outboundSchema.optional(),
    scoreThreshold: z.number().optional(),
}).transform((v) => {
    return remap$(v, {
        scoreThreshold: "score_threshold",
    });
});
export function rankingOptionsToJSON(rankingOptions) {
    return JSON.stringify(RankingOptions$outboundSchema.parse(rankingOptions));
}
export function rankingOptionsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RankingOptions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RankingOptions' from JSON`);
}
/** @internal */
export const OpenResponsesFileSearchTool$inboundSchema = z.object({
    type: z.literal("file_search"),
    vector_store_ids: z.array(z.string()),
    filters: z.nullable(z.union([
        z.lazy(() => OpenResponsesFileSearchToolFilters$inboundSchema),
        CompoundFilter$inboundSchema,
        z.any(),
    ])).optional(),
    max_num_results: z.int().optional(),
    ranking_options: z.lazy(() => RankingOptions$inboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        "vector_store_ids": "vectorStoreIds",
        "max_num_results": "maxNumResults",
        "ranking_options": "rankingOptions",
    });
});
/** @internal */
export const OpenResponsesFileSearchTool$outboundSchema = z.object({
    type: z.literal("file_search"),
    vectorStoreIds: z.array(z.string()),
    filters: z.nullable(z.union([
        z.lazy(() => OpenResponsesFileSearchToolFilters$outboundSchema),
        CompoundFilter$outboundSchema,
        z.any(),
    ])).optional(),
    maxNumResults: z.int().optional(),
    rankingOptions: z.lazy(() => RankingOptions$outboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        vectorStoreIds: "vector_store_ids",
        maxNumResults: "max_num_results",
        rankingOptions: "ranking_options",
    });
});
export function openResponsesFileSearchToolToJSON(openResponsesFileSearchTool) {
    return JSON.stringify(OpenResponsesFileSearchTool$outboundSchema.parse(openResponsesFileSearchTool));
}
export function openResponsesFileSearchToolFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesFileSearchTool$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesFileSearchTool' from JSON`);
}
