// @openrouter/sdk/esm/models/openresponseswebsearchpreviewtool.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 7d594dd37b72
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
import { ResponsesSearchContextSize$inboundSchema, ResponsesSearchContextSize$outboundSchema, } from "./responsessearchcontextsize.js";
import { WebSearchPreviewToolUserLocation$inboundSchema, WebSearchPreviewToolUserLocation$outboundSchema, } from "./websearchpreviewtooluserlocation.js";
/**
 * Which search engine to use. "auto" (default) uses native if the provider supports it, otherwise Exa. "native" forces the provider's built-in search. "exa" forces the Exa search API.
 */
export const OpenResponsesWebSearchPreviewToolEngine = {
    Auto: "auto",
    Native: "native",
    Exa: "exa",
};
/** @internal */
export const OpenResponsesWebSearchPreviewToolEngine$inboundSchema = openEnums.inboundSchema(OpenResponsesWebSearchPreviewToolEngine);
/** @internal */
export const OpenResponsesWebSearchPreviewToolEngine$outboundSchema = openEnums.outboundSchema(OpenResponsesWebSearchPreviewToolEngine);
/** @internal */
export const OpenResponsesWebSearchPreviewToolFilters$inboundSchema = z.object({
    allowed_domains: z.nullable(z.array(z.string())).optional(),
    excluded_domains: z.nullable(z.array(z.string())).optional(),
}).transform((v) => {
    return remap$(v, {
        "allowed_domains": "allowedDomains",
        "excluded_domains": "excludedDomains",
    });
});
/** @internal */
export const OpenResponsesWebSearchPreviewToolFilters$outboundSchema = z.object({
    allowedDomains: z.nullable(z.array(z.string())).optional(),
    excludedDomains: z.nullable(z.array(z.string())).optional(),
}).transform((v) => {
    return remap$(v, {
        allowedDomains: "allowed_domains",
        excludedDomains: "excluded_domains",
    });
});
export function openResponsesWebSearchPreviewToolFiltersToJSON(openResponsesWebSearchPreviewToolFilters) {
    return JSON.stringify(OpenResponsesWebSearchPreviewToolFilters$outboundSchema.parse(openResponsesWebSearchPreviewToolFilters));
}
export function openResponsesWebSearchPreviewToolFiltersFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesWebSearchPreviewToolFilters$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesWebSearchPreviewToolFilters' from JSON`);
}
/** @internal */
export const OpenResponsesWebSearchPreviewTool$inboundSchema = z.object({
    type: z.literal("web_search_preview"),
    search_context_size: ResponsesSearchContextSize$inboundSchema.optional(),
    user_location: z.nullable(WebSearchPreviewToolUserLocation$inboundSchema)
        .optional(),
    engine: OpenResponsesWebSearchPreviewToolEngine$inboundSchema.optional(),
    max_results: z.number().optional(),
    filters: z.nullable(z.lazy(() => OpenResponsesWebSearchPreviewToolFilters$inboundSchema)).optional(),
}).transform((v) => {
    return remap$(v, {
        "search_context_size": "searchContextSize",
        "user_location": "userLocation",
        "max_results": "maxResults",
    });
});
/** @internal */
export const OpenResponsesWebSearchPreviewTool$outboundSchema = z.object({
    type: z.literal("web_search_preview"),
    searchContextSize: ResponsesSearchContextSize$outboundSchema.optional(),
    userLocation: z.nullable(WebSearchPreviewToolUserLocation$outboundSchema)
        .optional(),
    engine: OpenResponsesWebSearchPreviewToolEngine$outboundSchema.optional(),
    maxResults: z.number().optional(),
    filters: z.nullable(z.lazy(() => OpenResponsesWebSearchPreviewToolFilters$outboundSchema)).optional(),
}).transform((v) => {
    return remap$(v, {
        searchContextSize: "search_context_size",
        userLocation: "user_location",
        maxResults: "max_results",
    });
});
export function openResponsesWebSearchPreviewToolToJSON(openResponsesWebSearchPreviewTool) {
    return JSON.stringify(OpenResponsesWebSearchPreviewTool$outboundSchema.parse(openResponsesWebSearchPreviewTool));
}
export function openResponsesWebSearchPreviewToolFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesWebSearchPreviewTool$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesWebSearchPreviewTool' from JSON`);
}
