// @openrouter/sdk/esm/models/openresponsesinputunion.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: b2182cf1a268
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import * as openEnums from "../types/enums.js";
import { OpenAIResponsesRefusalContent$outboundSchema, } from "./openairesponsesrefusalcontent.js";
import { OpenResponsesEasyInputMessage$outboundSchema, } from "./openresponseseasyinputmessage.js";
import { OpenResponsesFunctionCallOutput$outboundSchema, } from "./openresponsesfunctioncalloutput.js";
import { OpenResponsesFunctionToolCall$outboundSchema, } from "./openresponsesfunctiontoolcall.js";
import { OpenResponsesInputMessageItem$outboundSchema, } from "./openresponsesinputmessageitem.js";
import { OpenResponsesReasoning$outboundSchema, } from "./openresponsesreasoning.js";
import { ReasoningSummaryText$outboundSchema, } from "./reasoningsummarytext.js";
import { ReasoningTextContent$outboundSchema, } from "./reasoningtextcontent.js";
import { ResponseOutputText$outboundSchema, } from "./responseoutputtext.js";
import { ResponsesImageGenerationCall$outboundSchema, } from "./responsesimagegenerationcall.js";
import { ResponsesOutputItemFileSearchCall$outboundSchema, } from "./responsesoutputitemfilesearchcall.js";
import { ResponsesOutputItemFunctionCall$outboundSchema, } from "./responsesoutputitemfunctioncall.js";
import { ResponsesServerToolOutput$outboundSchema, } from "./responsesservertooloutput.js";
import { ResponsesWebSearchCallOutput$outboundSchema, } from "./responseswebsearchcalloutput.js";
export const OpenResponsesInputTypeReasoning = {
    Reasoning: "reasoning",
};
export const OpenResponsesInputStatusInProgress2 = {
    InProgress: "in_progress",
};
export const OpenResponsesInputStatusIncomplete2 = {
    Incomplete: "incomplete",
};
export const OpenResponsesInputStatusCompleted2 = {
    Completed: "completed",
};
/**
 * The format of the reasoning content
 */
export const OpenResponsesInputFormat = {
    Unknown: "unknown",
    OpenaiResponsesV1: "openai-responses-v1",
    AzureOpenaiResponsesV1: "azure-openai-responses-v1",
    XaiResponsesV1: "xai-responses-v1",
    AnthropicClaudeV1: "anthropic-claude-v1",
    GoogleGeminiV1: "google-gemini-v1",
};
export const OpenResponsesInputRole = {
    Assistant: "assistant",
};
export const OpenResponsesInputTypeMessage = {
    Message: "message",
};
export const OpenResponsesInputStatusInProgress1 = {
    InProgress: "in_progress",
};
export const OpenResponsesInputStatusIncomplete1 = {
    Incomplete: "incomplete",
};
export const OpenResponsesInputStatusCompleted1 = {
    Completed: "completed",
};
export const OpenResponsesInputPhaseFinalAnswer = {
    FinalAnswer: "final_answer",
};
export const OpenResponsesInputPhaseCommentary = {
    Commentary: "commentary",
};
/** @internal */
export const OpenResponsesInputTypeReasoning$outboundSchema = z.enum(OpenResponsesInputTypeReasoning);
/** @internal */
export const OpenResponsesInputStatusInProgress2$outboundSchema = z.enum(OpenResponsesInputStatusInProgress2);
/** @internal */
export const OpenResponsesInputStatusIncomplete2$outboundSchema = z.enum(OpenResponsesInputStatusIncomplete2);
/** @internal */
export const OpenResponsesInputStatusCompleted2$outboundSchema = z.enum(OpenResponsesInputStatusCompleted2);
/** @internal */
export const OpenResponsesInputStatusUnion2$outboundSchema = z.union([
    OpenResponsesInputStatusCompleted2$outboundSchema,
    OpenResponsesInputStatusIncomplete2$outboundSchema,
    OpenResponsesInputStatusInProgress2$outboundSchema,
]);
export function openResponsesInputStatusUnion2ToJSON(openResponsesInputStatusUnion2) {
    return JSON.stringify(OpenResponsesInputStatusUnion2$outboundSchema.parse(openResponsesInputStatusUnion2));
}
/** @internal */
export const OpenResponsesInputFormat$outboundSchema = openEnums.outboundSchema(OpenResponsesInputFormat);
/** @internal */
export const OpenResponsesInputReasoning$outboundSchema = z.object({
    type: OpenResponsesInputTypeReasoning$outboundSchema,
    id: z.string(),
    content: z.nullable(z.array(ReasoningTextContent$outboundSchema)).optional(),
    summary: z.nullable(z.array(ReasoningSummaryText$outboundSchema)),
    encryptedContent: z.nullable(z.string()).optional(),
    status: z.union([
        OpenResponsesInputStatusCompleted2$outboundSchema,
        OpenResponsesInputStatusIncomplete2$outboundSchema,
        OpenResponsesInputStatusInProgress2$outboundSchema,
    ]).optional(),
    signature: z.nullable(z.string()).optional(),
    format: z.nullable(OpenResponsesInputFormat$outboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        encryptedContent: "encrypted_content",
    });
});
export function openResponsesInputReasoningToJSON(openResponsesInputReasoning) {
    return JSON.stringify(OpenResponsesInputReasoning$outboundSchema.parse(openResponsesInputReasoning));
}
/** @internal */
export const OpenResponsesInputRole$outboundSchema = z.enum(OpenResponsesInputRole);
/** @internal */
export const OpenResponsesInputTypeMessage$outboundSchema = z.enum(OpenResponsesInputTypeMessage);
/** @internal */
export const OpenResponsesInputStatusInProgress1$outboundSchema = z.enum(OpenResponsesInputStatusInProgress1);
/** @internal */
export const OpenResponsesInputStatusIncomplete1$outboundSchema = z.enum(OpenResponsesInputStatusIncomplete1);
/** @internal */
export const OpenResponsesInputStatusCompleted1$outboundSchema = z.enum(OpenResponsesInputStatusCompleted1);
/** @internal */
export const OpenResponsesInputStatusUnion1$outboundSchema = z.union([
    OpenResponsesInputStatusCompleted1$outboundSchema,
    OpenResponsesInputStatusIncomplete1$outboundSchema,
    OpenResponsesInputStatusInProgress1$outboundSchema,
]);
export function openResponsesInputStatusUnion1ToJSON(openResponsesInputStatusUnion1) {
    return JSON.stringify(OpenResponsesInputStatusUnion1$outboundSchema.parse(openResponsesInputStatusUnion1));
}
/** @internal */
export const OpenResponsesInputContent1$outboundSchema = z.union([
    ResponseOutputText$outboundSchema,
    OpenAIResponsesRefusalContent$outboundSchema,
]);
export function openResponsesInputContent1ToJSON(openResponsesInputContent1) {
    return JSON.stringify(OpenResponsesInputContent1$outboundSchema.parse(openResponsesInputContent1));
}
/** @internal */
export const OpenResponsesInputContent2$outboundSchema = z.union([
    z.array(z.union([
        ResponseOutputText$outboundSchema,
        OpenAIResponsesRefusalContent$outboundSchema,
    ])),
    z.string(),
    z.any(),
]);
export function openResponsesInputContent2ToJSON(openResponsesInputContent2) {
    return JSON.stringify(OpenResponsesInputContent2$outboundSchema.parse(openResponsesInputContent2));
}
/** @internal */
export const OpenResponsesInputPhaseFinalAnswer$outboundSchema = z.enum(OpenResponsesInputPhaseFinalAnswer);
/** @internal */
export const OpenResponsesInputPhaseCommentary$outboundSchema = z.enum(OpenResponsesInputPhaseCommentary);
/** @internal */
export const OpenResponsesInputPhaseUnion$outboundSchema = z.union([
    OpenResponsesInputPhaseCommentary$outboundSchema,
    OpenResponsesInputPhaseFinalAnswer$outboundSchema,
    z.any(),
]);
export function openResponsesInputPhaseUnionToJSON(openResponsesInputPhaseUnion) {
    return JSON.stringify(OpenResponsesInputPhaseUnion$outboundSchema.parse(openResponsesInputPhaseUnion));
}
/** @internal */
export const OpenResponsesInputMessage$outboundSchema = z.object({
    id: z.string(),
    role: OpenResponsesInputRole$outboundSchema,
    type: OpenResponsesInputTypeMessage$outboundSchema,
    status: z.union([
        OpenResponsesInputStatusCompleted1$outboundSchema,
        OpenResponsesInputStatusIncomplete1$outboundSchema,
        OpenResponsesInputStatusInProgress1$outboundSchema,
    ]).optional(),
    content: z.nullable(z.union([
        z.array(z.union([
            ResponseOutputText$outboundSchema,
            OpenAIResponsesRefusalContent$outboundSchema,
        ])),
        z.string(),
        z.any(),
    ])).optional(),
    phase: z.nullable(z.union([
        OpenResponsesInputPhaseCommentary$outboundSchema,
        OpenResponsesInputPhaseFinalAnswer$outboundSchema,
        z.any(),
    ])).optional(),
});
export function openResponsesInputMessageToJSON(openResponsesInputMessage) {
    return JSON.stringify(OpenResponsesInputMessage$outboundSchema.parse(openResponsesInputMessage));
}
/** @internal */
export const OpenResponsesInputUnion1$outboundSchema = z.union([
    OpenResponsesFunctionToolCall$outboundSchema,
    z.lazy(() => OpenResponsesInputMessage$outboundSchema),
    ResponsesOutputItemFunctionCall$outboundSchema,
    ResponsesWebSearchCallOutput$outboundSchema,
    ResponsesOutputItemFileSearchCall$outboundSchema,
    OpenResponsesReasoning$outboundSchema,
    OpenResponsesFunctionCallOutput$outboundSchema,
    z.lazy(() => OpenResponsesInputReasoning$outboundSchema),
    ResponsesImageGenerationCall$outboundSchema,
    ResponsesServerToolOutput$outboundSchema,
    OpenResponsesEasyInputMessage$outboundSchema,
    OpenResponsesInputMessageItem$outboundSchema,
]);
export function openResponsesInputUnion1ToJSON(openResponsesInputUnion1) {
    return JSON.stringify(OpenResponsesInputUnion1$outboundSchema.parse(openResponsesInputUnion1));
}
/** @internal */
export const OpenResponsesInputUnion$outboundSchema = z.union([
    z.string(),
    z.array(z.union([
        OpenResponsesFunctionToolCall$outboundSchema,
        z.lazy(() => OpenResponsesInputMessage$outboundSchema),
        ResponsesOutputItemFunctionCall$outboundSchema,
        ResponsesWebSearchCallOutput$outboundSchema,
        ResponsesOutputItemFileSearchCall$outboundSchema,
        OpenResponsesReasoning$outboundSchema,
        OpenResponsesFunctionCallOutput$outboundSchema,
        z.lazy(() => OpenResponsesInputReasoning$outboundSchema),
        ResponsesImageGenerationCall$outboundSchema,
        ResponsesServerToolOutput$outboundSchema,
        OpenResponsesEasyInputMessage$outboundSchema,
        OpenResponsesInputMessageItem$outboundSchema,
    ])),
]);
export function openResponsesInputUnionToJSON(openResponsesInputUnion) {
    return JSON.stringify(OpenResponsesInputUnion$outboundSchema.parse(openResponsesInputUnion));
}
