// @openrouter/sdk/esm/models/openresponsescodeinterpretertool.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 8d6c57413f99
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
export const ContainerType = {
    Auto: "auto",
};
export const MemoryLimit = {
    Oneg: "1g",
    Fourg: "4g",
    Sixteeng: "16g",
    SixtyFourg: "64g",
};
/** @internal */
export const ContainerType$inboundSchema = z
    .enum(ContainerType);
/** @internal */
export const ContainerType$outboundSchema = ContainerType$inboundSchema;
/** @internal */
export const MemoryLimit$inboundSchema = openEnums.inboundSchema(MemoryLimit);
/** @internal */
export const MemoryLimit$outboundSchema = openEnums.outboundSchema(MemoryLimit);
/** @internal */
export const ContainerAuto$inboundSchema = z
    .object({
    type: ContainerType$inboundSchema,
    file_ids: z.array(z.string()).optional(),
    memory_limit: z.nullable(MemoryLimit$inboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        "file_ids": "fileIds",
        "memory_limit": "memoryLimit",
    });
});
/** @internal */
export const ContainerAuto$outboundSchema = z.object({
    type: ContainerType$outboundSchema,
    fileIds: z.array(z.string()).optional(),
    memoryLimit: z.nullable(MemoryLimit$outboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        fileIds: "file_ids",
        memoryLimit: "memory_limit",
    });
});
export function containerAutoToJSON(containerAuto) {
    return JSON.stringify(ContainerAuto$outboundSchema.parse(containerAuto));
}
export function containerAutoFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ContainerAuto$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ContainerAuto' from JSON`);
}
/** @internal */
export const Container$inboundSchema = z.union([
    z.lazy(() => ContainerAuto$inboundSchema),
    z.string(),
]);
/** @internal */
export const Container$outboundSchema = z.union([z.lazy(() => ContainerAuto$outboundSchema), z.string()]);
export function containerToJSON(container) {
    return JSON.stringify(Container$outboundSchema.parse(container));
}
export function containerFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Container$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Container' from JSON`);
}
/** @internal */
export const OpenResponsesCodeInterpreterTool$inboundSchema = z.object({
    type: z.literal("code_interpreter"),
    container: z.union([z.lazy(() => ContainerAuto$inboundSchema), z.string()]),
});
/** @internal */
export const OpenResponsesCodeInterpreterTool$outboundSchema = z.object({
    type: z.literal("code_interpreter"),
    container: z.union([z.lazy(() => ContainerAuto$outboundSchema), z.string()]),
});
export function openResponsesCodeInterpreterToolToJSON(openResponsesCodeInterpreterTool) {
    return JSON.stringify(OpenResponsesCodeInterpreterTool$outboundSchema.parse(openResponsesCodeInterpreterTool));
}
export function openResponsesCodeInterpreterToolFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesCodeInterpreterTool$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesCodeInterpreterTool' from JSON`);
}
