// @openrouter/sdk/esm/models/providerpreferences.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 3a47295d7b91
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import * as openEnums from "../types/enums.js";
import { DataCollection$outboundSchema, } from "./datacollection.js";
import { PreferredMaxLatency$outboundSchema, } from "./preferredmaxlatency.js";
import { PreferredMinThroughput$outboundSchema, } from "./preferredminthroughput.js";
import { ProviderName$outboundSchema } from "./providername.js";
import { Quantization$outboundSchema } from "./quantization.js";
export const ProviderPreferencesSortEnum = {
    Price: "price",
    Throughput: "throughput",
    Latency: "latency",
    Exacto: "exacto",
};
export const ProviderPreferencesProviderSortConfigEnum = {
    Price: "price",
    Throughput: "throughput",
    Latency: "latency",
    Exacto: "exacto",
};
/**
 * The provider sorting strategy (price, throughput, latency)
 */
export const ProviderPreferencesBy = {
    Price: "price",
    Throughput: "throughput",
    Latency: "latency",
    Exacto: "exacto",
};
/**
 * Partitioning strategy for sorting: "model" (default) groups endpoints by model before sorting (fallback models remain fallbacks), "none" sorts all endpoints together regardless of model.
 */
export const ProviderPreferencesPartition = {
    Model: "model",
    None: "none",
};
/**
 * The provider sorting strategy (price, throughput, latency)
 */
export const ProviderPreferencesProviderSort = {
    Price: "price",
    Throughput: "throughput",
    Latency: "latency",
    Exacto: "exacto",
};
/** @internal */
export const ProviderPreferencesOrder$outboundSchema = z.union([ProviderName$outboundSchema, z.string()]);
export function providerPreferencesOrderToJSON(providerPreferencesOrder) {
    return JSON.stringify(ProviderPreferencesOrder$outboundSchema.parse(providerPreferencesOrder));
}
/** @internal */
export const ProviderPreferencesOnly$outboundSchema = z.union([ProviderName$outboundSchema, z.string()]);
export function providerPreferencesOnlyToJSON(providerPreferencesOnly) {
    return JSON.stringify(ProviderPreferencesOnly$outboundSchema.parse(providerPreferencesOnly));
}
/** @internal */
export const ProviderPreferencesIgnore$outboundSchema = z.union([ProviderName$outboundSchema, z.string()]);
export function providerPreferencesIgnoreToJSON(providerPreferencesIgnore) {
    return JSON.stringify(ProviderPreferencesIgnore$outboundSchema.parse(providerPreferencesIgnore));
}
/** @internal */
export const ProviderPreferencesSortEnum$outboundSchema = openEnums.outboundSchema(ProviderPreferencesSortEnum);
/** @internal */
export const ProviderPreferencesProviderSortConfigEnum$outboundSchema = z.enum(ProviderPreferencesProviderSortConfigEnum);
/** @internal */
export const ProviderPreferencesBy$outboundSchema = openEnums.outboundSchema(ProviderPreferencesBy);
/** @internal */
export const ProviderPreferencesPartition$outboundSchema = openEnums.outboundSchema(ProviderPreferencesPartition);
/** @internal */
export const ProviderPreferencesProviderSortConfig$outboundSchema = z.object({
    by: z.nullable(ProviderPreferencesBy$outboundSchema).optional(),
    partition: z.nullable(ProviderPreferencesPartition$outboundSchema).optional(),
});
export function providerPreferencesProviderSortConfigToJSON(providerPreferencesProviderSortConfig) {
    return JSON.stringify(ProviderPreferencesProviderSortConfig$outboundSchema.parse(providerPreferencesProviderSortConfig));
}
/** @internal */
export const ProviderPreferencesProviderSortConfigUnion$outboundSchema = z.union([
    z.lazy(() => ProviderPreferencesProviderSortConfig$outboundSchema),
    ProviderPreferencesProviderSortConfigEnum$outboundSchema,
]);
export function providerPreferencesProviderSortConfigUnionToJSON(providerPreferencesProviderSortConfigUnion) {
    return JSON.stringify(ProviderPreferencesProviderSortConfigUnion$outboundSchema.parse(providerPreferencesProviderSortConfigUnion));
}
/** @internal */
export const ProviderPreferencesProviderSort$outboundSchema = openEnums.outboundSchema(ProviderPreferencesProviderSort);
/** @internal */
export const ProviderPreferencesSortUnion$outboundSchema = z.union([
    ProviderPreferencesProviderSort$outboundSchema,
    z.union([
        z.lazy(() => ProviderPreferencesProviderSortConfig$outboundSchema),
        ProviderPreferencesProviderSortConfigEnum$outboundSchema,
    ]),
    ProviderPreferencesSortEnum$outboundSchema,
]);
export function providerPreferencesSortUnionToJSON(providerPreferencesSortUnion) {
    return JSON.stringify(ProviderPreferencesSortUnion$outboundSchema.parse(providerPreferencesSortUnion));
}
/** @internal */
export const ProviderPreferencesMaxPrice$outboundSchema = z.object({
    prompt: z.string().optional(),
    completion: z.string().optional(),
    image: z.string().optional(),
    audio: z.string().optional(),
    request: z.string().optional(),
});
export function providerPreferencesMaxPriceToJSON(providerPreferencesMaxPrice) {
    return JSON.stringify(ProviderPreferencesMaxPrice$outboundSchema.parse(providerPreferencesMaxPrice));
}
/** @internal */
export const ProviderPreferences$outboundSchema = z.object({
    allowFallbacks: z.nullable(z.boolean()).optional(),
    requireParameters: z.nullable(z.boolean()).optional(),
    dataCollection: z.nullable(DataCollection$outboundSchema).optional(),
    zdr: z.nullable(z.boolean()).optional(),
    enforceDistillableText: z.nullable(z.boolean()).optional(),
    order: z.nullable(z.array(z.union([ProviderName$outboundSchema, z.string()])))
        .optional(),
    only: z.nullable(z.array(z.union([ProviderName$outboundSchema, z.string()])))
        .optional(),
    ignore: z.nullable(z.array(z.union([ProviderName$outboundSchema, z.string()]))).optional(),
    quantizations: z.nullable(z.array(Quantization$outboundSchema)).optional(),
    sort: z.nullable(z.union([
        ProviderPreferencesProviderSort$outboundSchema,
        z.union([
            z.lazy(() => ProviderPreferencesProviderSortConfig$outboundSchema),
            ProviderPreferencesProviderSortConfigEnum$outboundSchema,
        ]),
        ProviderPreferencesSortEnum$outboundSchema,
    ])).optional(),
    maxPrice: z.lazy(() => ProviderPreferencesMaxPrice$outboundSchema).optional(),
    preferredMinThroughput: z.nullable(PreferredMinThroughput$outboundSchema)
        .optional(),
    preferredMaxLatency: z.nullable(PreferredMaxLatency$outboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        allowFallbacks: "allow_fallbacks",
        requireParameters: "require_parameters",
        dataCollection: "data_collection",
        enforceDistillableText: "enforce_distillable_text",
        maxPrice: "max_price",
        preferredMinThroughput: "preferred_min_throughput",
        preferredMaxLatency: "preferred_max_latency",
    });
});
export function providerPreferencesToJSON(providerPreferences) {
    return JSON.stringify(ProviderPreferences$outboundSchema.parse(providerPreferences));
}
