// @openrouter/sdk/esm/models/reasoningdetailencrypted.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 2064e7820eab
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
export const ReasoningDetailEncryptedFormat = {
    Unknown: "unknown",
    OpenaiResponsesV1: "openai-responses-v1",
    AzureOpenaiResponsesV1: "azure-openai-responses-v1",
    XaiResponsesV1: "xai-responses-v1",
    AnthropicClaudeV1: "anthropic-claude-v1",
    GoogleGeminiV1: "google-gemini-v1",
};
/** @internal */
export const ReasoningDetailEncryptedFormat$inboundSchema = openEnums.inboundSchema(ReasoningDetailEncryptedFormat);
/** @internal */
export const ReasoningDetailEncryptedFormat$outboundSchema = openEnums.outboundSchema(ReasoningDetailEncryptedFormat);
/** @internal */
export const ReasoningDetailEncrypted$inboundSchema = z.object({
    type: z.literal("reasoning.encrypted"),
    data: z.string(),
    id: z.nullable(z.string()).optional(),
    format: z.nullable(ReasoningDetailEncryptedFormat$inboundSchema).optional(),
    index: z.number().optional(),
});
/** @internal */
export const ReasoningDetailEncrypted$outboundSchema = z.object({
    type: z.literal("reasoning.encrypted"),
    data: z.string(),
    id: z.nullable(z.string()).optional(),
    format: z.nullable(ReasoningDetailEncryptedFormat$outboundSchema).optional(),
    index: z.number().optional(),
});
export function reasoningDetailEncryptedToJSON(reasoningDetailEncrypted) {
    return JSON.stringify(ReasoningDetailEncrypted$outboundSchema.parse(reasoningDetailEncrypted));
}
export function reasoningDetailEncryptedFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ReasoningDetailEncrypted$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ReasoningDetailEncrypted' from JSON`);
}
