// @openrouter/sdk/esm/models/reasoningdetailsummary.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 08e5d256b8c9
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
export const ReasoningDetailSummaryFormat = {
    Unknown: "unknown",
    OpenaiResponsesV1: "openai-responses-v1",
    AzureOpenaiResponsesV1: "azure-openai-responses-v1",
    XaiResponsesV1: "xai-responses-v1",
    AnthropicClaudeV1: "anthropic-claude-v1",
    GoogleGeminiV1: "google-gemini-v1",
};
/** @internal */
export const ReasoningDetailSummaryFormat$inboundSchema = openEnums.inboundSchema(ReasoningDetailSummaryFormat);
/** @internal */
export const ReasoningDetailSummaryFormat$outboundSchema = openEnums.outboundSchema(ReasoningDetailSummaryFormat);
/** @internal */
export const ReasoningDetailSummary$inboundSchema = z.object({
    type: z.literal("reasoning.summary"),
    summary: z.string(),
    id: z.nullable(z.string()).optional(),
    format: z.nullable(ReasoningDetailSummaryFormat$inboundSchema).optional(),
    index: z.number().optional(),
});
/** @internal */
export const ReasoningDetailSummary$outboundSchema = z.object({
    type: z.literal("reasoning.summary"),
    summary: z.string(),
    id: z.nullable(z.string()).optional(),
    format: z.nullable(ReasoningDetailSummaryFormat$outboundSchema).optional(),
    index: z.number().optional(),
});
export function reasoningDetailSummaryToJSON(reasoningDetailSummary) {
    return JSON.stringify(ReasoningDetailSummary$outboundSchema.parse(reasoningDetailSummary));
}
export function reasoningDetailSummaryFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ReasoningDetailSummary$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ReasoningDetailSummary' from JSON`);
}
