// @openrouter/sdk/esm/models/reasoningdetailtext.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 8b6bd4c64a66
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
export const ReasoningDetailTextFormat = {
    Unknown: "unknown",
    OpenaiResponsesV1: "openai-responses-v1",
    AzureOpenaiResponsesV1: "azure-openai-responses-v1",
    XaiResponsesV1: "xai-responses-v1",
    AnthropicClaudeV1: "anthropic-claude-v1",
    GoogleGeminiV1: "google-gemini-v1",
};
/** @internal */
export const ReasoningDetailTextFormat$inboundSchema = openEnums.inboundSchema(ReasoningDetailTextFormat);
/** @internal */
export const ReasoningDetailTextFormat$outboundSchema = openEnums.outboundSchema(ReasoningDetailTextFormat);
/** @internal */
export const ReasoningDetailText$inboundSchema = z.object({
    type: z.literal("reasoning.text"),
    text: z.nullable(z.string()).optional(),
    signature: z.nullable(z.string()).optional(),
    id: z.nullable(z.string()).optional(),
    format: z.nullable(ReasoningDetailTextFormat$inboundSchema).optional(),
    index: z.number().optional(),
});
/** @internal */
export const ReasoningDetailText$outboundSchema = z.object({
    type: z.literal("reasoning.text"),
    text: z.nullable(z.string()).optional(),
    signature: z.nullable(z.string()).optional(),
    id: z.nullable(z.string()).optional(),
    format: z.nullable(ReasoningDetailTextFormat$outboundSchema).optional(),
    index: z.number().optional(),
});
export function reasoningDetailTextToJSON(reasoningDetailText) {
    return JSON.stringify(ReasoningDetailText$outboundSchema.parse(reasoningDetailText));
}
export function reasoningDetailTextFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ReasoningDetailText$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ReasoningDetailText' from JSON`);
}
