// @openrouter/sdk/esm/models/responseswebsearchservertool.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 8ef0e49c5731
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
/** @internal */
export const ResponsesWebSearchServerToolParameters$outboundSchema = z.object({
    maxResults: z.number().optional(),
    maxTotalResults: z.number().optional(),
}).transform((v) => {
    return remap$(v, {
        maxResults: "max_results",
        maxTotalResults: "max_total_results",
    });
});
export function responsesWebSearchServerToolParametersToJSON(responsesWebSearchServerToolParameters) {
    return JSON.stringify(ResponsesWebSearchServerToolParameters$outboundSchema.parse(responsesWebSearchServerToolParameters));
}
/** @internal */
export const ResponsesWebSearchServerTool$outboundSchema = z.object({
    type: z.literal("openrouter:web_search"),
    parameters: z.lazy(() => ResponsesWebSearchServerToolParameters$outboundSchema).optional(),
});
export function responsesWebSearchServerToolToJSON(responsesWebSearchServerTool) {
    return JSON.stringify(ResponsesWebSearchServerTool$outboundSchema.parse(responsesWebSearchServerTool));
}
