// @openrouter/sdk/esm/models/responsesservertooloutput.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 40768491f00e
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { collectExtraKeys as collectExtraKeys$, safeParse, } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
export const ResponsesServerToolOutputStatus = {
    Completed: "completed",
    InProgress: "in_progress",
    Incomplete: "incomplete",
};
/** @internal */
export const ResponsesServerToolOutputStatus$inboundSchema = openEnums.inboundSchema(ResponsesServerToolOutputStatus);
/** @internal */
export const ResponsesServerToolOutputStatus$outboundSchema = openEnums.outboundSchema(ResponsesServerToolOutputStatus);
/** @internal */
export const ResponsesServerToolOutput$inboundSchema = collectExtraKeys$(z.object({
    type: z.string(),
    id: z.string().optional(),
    status: ResponsesServerToolOutputStatus$inboundSchema,
}).catchall(z.any()), "additionalProperties", true);
/** @internal */
export const ResponsesServerToolOutput$outboundSchema = z.object({
    type: z.string(),
    id: z.string().optional(),
    status: ResponsesServerToolOutputStatus$outboundSchema,
    additionalProperties: z.record(z.string(), z.nullable(z.any())).optional(),
}).transform((v) => {
    return {
        ...v.additionalProperties,
        ...remap$(v, {
            additionalProperties: null,
        }),
    };
});
export function responsesServerToolOutputToJSON(responsesServerToolOutput) {
    return JSON.stringify(ResponsesServerToolOutput$outboundSchema.parse(responsesServerToolOutput));
}
export function responsesServerToolOutputFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ResponsesServerToolOutput$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ResponsesServerToolOutput' from JSON`);
}
