// @openrouter/sdk/esm/models/responsesoutputmessage.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 32cb33488ea2
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import { OpenAIResponsesRefusalContent$inboundSchema, } from "./openairesponsesrefusalcontent.js";
import { ResponseOutputText$inboundSchema, } from "./responseoutputtext.js";
export const ResponsesOutputMessageRole = {
    Assistant: "assistant",
};
export const ResponsesOutputMessageType = {
    Message: "message",
};
export const ResponsesOutputMessageStatusInProgress = {
    InProgress: "in_progress",
};
export const ResponsesOutputMessageStatusIncomplete = {
    Incomplete: "incomplete",
};
export const ResponsesOutputMessageStatusCompleted = {
    Completed: "completed",
};
export const ResponsesOutputMessagePhaseFinalAnswer = {
    FinalAnswer: "final_answer",
};
export const ResponsesOutputMessagePhaseCommentary = {
    Commentary: "commentary",
};
/** @internal */
export const ResponsesOutputMessageRole$inboundSchema = z.enum(ResponsesOutputMessageRole);
/** @internal */
export const ResponsesOutputMessageType$inboundSchema = z.enum(ResponsesOutputMessageType);
/** @internal */
export const ResponsesOutputMessageStatusInProgress$inboundSchema = z.enum(ResponsesOutputMessageStatusInProgress);
/** @internal */
export const ResponsesOutputMessageStatusIncomplete$inboundSchema = z.enum(ResponsesOutputMessageStatusIncomplete);
/** @internal */
export const ResponsesOutputMessageStatusCompleted$inboundSchema = z.enum(ResponsesOutputMessageStatusCompleted);
/** @internal */
export const ResponsesOutputMessageStatusUnion$inboundSchema = z.union([
    ResponsesOutputMessageStatusCompleted$inboundSchema,
    ResponsesOutputMessageStatusIncomplete$inboundSchema,
    ResponsesOutputMessageStatusInProgress$inboundSchema,
]);
export function responsesOutputMessageStatusUnionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ResponsesOutputMessageStatusUnion$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ResponsesOutputMessageStatusUnion' from JSON`);
}
/** @internal */
export const ResponsesOutputMessageContent$inboundSchema = z.union([
    ResponseOutputText$inboundSchema,
    OpenAIResponsesRefusalContent$inboundSchema,
]);
export function responsesOutputMessageContentFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ResponsesOutputMessageContent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ResponsesOutputMessageContent' from JSON`);
}
/** @internal */
export const ResponsesOutputMessagePhaseFinalAnswer$inboundSchema = z.enum(ResponsesOutputMessagePhaseFinalAnswer);
/** @internal */
export const ResponsesOutputMessagePhaseCommentary$inboundSchema = z.enum(ResponsesOutputMessagePhaseCommentary);
/** @internal */
export const ResponsesOutputMessagePhaseUnion$inboundSchema = z.union([
    ResponsesOutputMessagePhaseCommentary$inboundSchema,
    ResponsesOutputMessagePhaseFinalAnswer$inboundSchema,
    z.any(),
]);
export function responsesOutputMessagePhaseUnionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ResponsesOutputMessagePhaseUnion$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ResponsesOutputMessagePhaseUnion' from JSON`);
}
/** @internal */
export const ResponsesOutputMessage$inboundSchema = z.object({
    id: z.string(),
    role: ResponsesOutputMessageRole$inboundSchema,
    type: ResponsesOutputMessageType$inboundSchema,
    status: z.union([
        ResponsesOutputMessageStatusCompleted$inboundSchema,
        ResponsesOutputMessageStatusIncomplete$inboundSchema,
        ResponsesOutputMessageStatusInProgress$inboundSchema,
    ]).optional(),
    content: z.array(z.union([
        ResponseOutputText$inboundSchema,
        OpenAIResponsesRefusalContent$inboundSchema,
    ])),
    phase: z.nullable(z.union([
        ResponsesOutputMessagePhaseCommentary$inboundSchema,
        ResponsesOutputMessagePhaseFinalAnswer$inboundSchema,
        z.any(),
    ])).optional(),
});
export function responsesOutputMessageFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ResponsesOutputMessage$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ResponsesOutputMessage' from JSON`);
}
