// @openrouter/sdk/esm/models/responseswebsearchcalloutput.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: c1c0ab68e92c
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import { WebSearchStatus$inboundSchema, WebSearchStatus$outboundSchema, } from "./websearchstatus.js";
export const TypeWebSearchCall = {
    WebSearchCall: "web_search_call",
};
export const TypeURL = {
    Url: "url",
};
/** @internal */
export const TypeWebSearchCall$inboundSchema = z.enum(TypeWebSearchCall);
/** @internal */
export const TypeWebSearchCall$outboundSchema = TypeWebSearchCall$inboundSchema;
/** @internal */
export const ActionFindInPage$inboundSchema = z.object({
    type: z.literal("find_in_page"),
    pattern: z.string(),
    url: z.string(),
});
/** @internal */
export const ActionFindInPage$outboundSchema = z.object({
    type: z.literal("find_in_page"),
    pattern: z.string(),
    url: z.string(),
});
export function actionFindInPageToJSON(actionFindInPage) {
    return JSON.stringify(ActionFindInPage$outboundSchema.parse(actionFindInPage));
}
export function actionFindInPageFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ActionFindInPage$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ActionFindInPage' from JSON`);
}
/** @internal */
export const ActionOpenPage$inboundSchema = z.object({
    type: z.literal("open_page"),
    url: z.nullable(z.string()).optional(),
});
/** @internal */
export const ActionOpenPage$outboundSchema = z.object({
    type: z.literal("open_page"),
    url: z.nullable(z.string()).optional(),
});
export function actionOpenPageToJSON(actionOpenPage) {
    return JSON.stringify(ActionOpenPage$outboundSchema.parse(actionOpenPage));
}
export function actionOpenPageFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ActionOpenPage$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ActionOpenPage' from JSON`);
}
/** @internal */
export const TypeURL$inboundSchema = z.enum(TypeURL);
/** @internal */
export const TypeURL$outboundSchema = TypeURL$inboundSchema;
/** @internal */
export const Source$inboundSchema = z.object({
    type: TypeURL$inboundSchema,
    url: z.string(),
});
/** @internal */
export const Source$outboundSchema = z
    .object({
    type: TypeURL$outboundSchema,
    url: z.string(),
});
export function sourceToJSON(source) {
    return JSON.stringify(Source$outboundSchema.parse(source));
}
export function sourceFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Source$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Source' from JSON`);
}
/** @internal */
export const ActionSearch$inboundSchema = z
    .object({
    type: z.literal("search"),
    query: z.string(),
    queries: z.array(z.string()).optional(),
    sources: z.array(z.lazy(() => Source$inboundSchema)).optional(),
});
/** @internal */
export const ActionSearch$outboundSchema = z.object({
    type: z.literal("search"),
    query: z.string(),
    queries: z.array(z.string()).optional(),
    sources: z.array(z.lazy(() => Source$outboundSchema)).optional(),
});
export function actionSearchToJSON(actionSearch) {
    return JSON.stringify(ActionSearch$outboundSchema.parse(actionSearch));
}
export function actionSearchFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ActionSearch$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ActionSearch' from JSON`);
}
/** @internal */
export const Action$inboundSchema = z.union([
    z.lazy(() => ActionSearch$inboundSchema),
    z.lazy(() => ActionOpenPage$inboundSchema),
    z.lazy(() => ActionFindInPage$inboundSchema),
]);
/** @internal */
export const Action$outboundSchema = z
    .union([
    z.lazy(() => ActionSearch$outboundSchema),
    z.lazy(() => ActionOpenPage$outboundSchema),
    z.lazy(() => ActionFindInPage$outboundSchema),
]);
export function actionToJSON(action) {
    return JSON.stringify(Action$outboundSchema.parse(action));
}
export function actionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Action$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Action' from JSON`);
}
/** @internal */
export const ResponsesWebSearchCallOutput$inboundSchema = z.object({
    type: TypeWebSearchCall$inboundSchema,
    id: z.string(),
    action: z.union([
        z.lazy(() => ActionSearch$inboundSchema),
        z.lazy(() => ActionOpenPage$inboundSchema),
        z.lazy(() => ActionFindInPage$inboundSchema),
    ]),
    status: WebSearchStatus$inboundSchema,
});
/** @internal */
export const ResponsesWebSearchCallOutput$outboundSchema = z.object({
    type: TypeWebSearchCall$outboundSchema,
    id: z.string(),
    action: z.union([
        z.lazy(() => ActionSearch$outboundSchema),
        z.lazy(() => ActionOpenPage$outboundSchema),
        z.lazy(() => ActionFindInPage$outboundSchema),
    ]),
    status: WebSearchStatus$outboundSchema,
});
export function responsesWebSearchCallOutputToJSON(responsesWebSearchCallOutput) {
    return JSON.stringify(ResponsesWebSearchCallOutput$outboundSchema.parse(responsesWebSearchCallOutput));
}
export function responsesWebSearchCallOutputFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ResponsesWebSearchCallOutput$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ResponsesWebSearchCallOutput' from JSON`);
}
