// @openrouter/sdk/esm/models/tooldefinitionjson.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 546c340f3013
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { ChatMessageContentItemCacheControl$outboundSchema, } from "./chatmessagecontentitemcachecontrol.js";
import { DatetimeServerTool$outboundSchema, } from "./datetimeservertool.js";
import { WebSearchServerTool$outboundSchema, } from "./websearchservertool.js";
import { WebSearchShorthand$outboundSchema, } from "./websearchshorthand.js";
/** @internal */
export const ToolDefinitionJsonFunctionFunction$outboundSchema = z.object({
    name: z.string(),
    description: z.string().optional(),
    parameters: z.record(z.string(), z.nullable(z.any())).optional(),
    strict: z.nullable(z.boolean()).optional(),
});
export function toolDefinitionJsonFunctionFunctionToJSON(toolDefinitionJsonFunctionFunction) {
    return JSON.stringify(ToolDefinitionJsonFunctionFunction$outboundSchema.parse(toolDefinitionJsonFunctionFunction));
}
/** @internal */
export const ToolDefinitionJsonFunction$outboundSchema = z.object({
    type: z.literal("function"),
    function: z.lazy(() => ToolDefinitionJsonFunctionFunction$outboundSchema),
    cacheControl: ChatMessageContentItemCacheControl$outboundSchema.optional(),
}).transform((v) => {
    return remap$(v, {
        cacheControl: "cache_control",
    });
});
export function toolDefinitionJsonFunctionToJSON(toolDefinitionJsonFunction) {
    return JSON.stringify(ToolDefinitionJsonFunction$outboundSchema.parse(toolDefinitionJsonFunction));
}
/** @internal */
export const ToolDefinitionJson$outboundSchema = z.union([
    z.lazy(() => ToolDefinitionJsonFunction$outboundSchema),
    DatetimeServerTool$outboundSchema,
    WebSearchServerTool$outboundSchema,
    WebSearchShorthand$outboundSchema.and(z.object({ type: z.literal("web_search") })),
    WebSearchShorthand$outboundSchema.and(z.object({ type: z.literal("web_search_preview") })),
    WebSearchShorthand$outboundSchema.and(z.object({ type: z.literal("web_search_preview_2025_03_11") })),
    WebSearchShorthand$outboundSchema.and(z.object({ type: z.literal("web_search_2025_08_26") })),
]);
export function toolDefinitionJsonToJSON(toolDefinitionJson) {
    return JSON.stringify(ToolDefinitionJson$outboundSchema.parse(toolDefinitionJson));
}
