// @openrouter/sdk/esm/models/toolchoiceoption.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: d767ca0ded82
 */
import * as z from "zod/v4";
import { NamedToolChoice$outboundSchema, } from "./namedtoolchoice.js";
export const ToolChoiceOptionRequired = {
    Required: "required",
};
export const ToolChoiceOptionAuto = {
    Auto: "auto",
};
export const ToolChoiceOptionNone = {
    None: "none",
};
/** @internal */
export const ToolChoiceOptionRequired$outboundSchema = z.enum(ToolChoiceOptionRequired);
/** @internal */
export const ToolChoiceOptionAuto$outboundSchema = z.enum(ToolChoiceOptionAuto);
/** @internal */
export const ToolChoiceOptionNone$outboundSchema = z.enum(ToolChoiceOptionNone);
/** @internal */
export const ToolChoiceOption$outboundSchema = z.union([
    NamedToolChoice$outboundSchema,
    ToolChoiceOptionNone$outboundSchema,
    ToolChoiceOptionAuto$outboundSchema,
    ToolChoiceOptionRequired$outboundSchema,
]);
export function toolChoiceOptionToJSON(toolChoiceOption) {
    return JSON.stringify(ToolChoiceOption$outboundSchema.parse(toolChoiceOption));
}
