// @openrouter/sdk/esm/models/websearchshorthand.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 1de0cb463e93
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import * as openEnums from "../types/enums.js";
export const WebSearchShorthandType = {
    WebSearch: "web_search",
    WebSearchPreview: "web_search_preview",
    WebSearchPreview20250311: "web_search_preview_2025_03_11",
    WebSearch20250826: "web_search_2025_08_26",
};
/**
 * Which search engine to use. "auto" (default) uses native if the provider supports it, otherwise Exa. "native" forces the provider's built-in search. "exa" forces the Exa search API.
 */
export const WebSearchShorthandEngine = {
    Auto: "auto",
    Native: "native",
    Exa: "exa",
};
/**
 * How much context to retrieve per result. Defaults to medium (15000 chars).
 */
export const WebSearchShorthandSearchContextSize = {
    Low: "low",
    Medium: "medium",
    High: "high",
};
export const WebSearchShorthandTypeApproximate = {
    Approximate: "approximate",
};
/**
 * Which search engine to use. "auto" (default) uses native if the provider supports it, otherwise Exa. "native" forces the provider's built-in search. "exa" forces the Exa search API.
 */
export const WebSearchShorthandParametersEngine = {
    Auto: "auto",
    Native: "native",
    Exa: "exa",
};
/**
 * How much context to retrieve per result. Defaults to medium (15000 chars).
 */
export const WebSearchShorthandParametersSearchContextSize = {
    Low: "low",
    Medium: "medium",
    High: "high",
};
export const WebSearchShorthandParametersType = {
    Approximate: "approximate",
};
/** @internal */
export const WebSearchShorthandType$outboundSchema = openEnums.outboundSchema(WebSearchShorthandType);
/** @internal */
export const WebSearchShorthandEngine$outboundSchema = openEnums.outboundSchema(WebSearchShorthandEngine);
/** @internal */
export const WebSearchShorthandSearchContextSize$outboundSchema = openEnums.outboundSchema(WebSearchShorthandSearchContextSize);
/** @internal */
export const WebSearchShorthandTypeApproximate$outboundSchema = z.enum(WebSearchShorthandTypeApproximate);
/** @internal */
export const WebSearchShorthandUserLocation$outboundSchema = z.object({
    type: WebSearchShorthandTypeApproximate$outboundSchema.optional(),
    city: z.string().optional(),
    region: z.string().optional(),
    country: z.string().optional(),
    timezone: z.string().optional(),
});
export function webSearchShorthandUserLocationToJSON(webSearchShorthandUserLocation) {
    return JSON.stringify(WebSearchShorthandUserLocation$outboundSchema.parse(webSearchShorthandUserLocation));
}
/** @internal */
export const WebSearchShorthandParametersEngine$outboundSchema = openEnums.outboundSchema(WebSearchShorthandParametersEngine);
/** @internal */
export const WebSearchShorthandParametersSearchContextSize$outboundSchema = openEnums
    .outboundSchema(WebSearchShorthandParametersSearchContextSize);
/** @internal */
export const WebSearchShorthandParametersType$outboundSchema = z.enum(WebSearchShorthandParametersType);
/** @internal */
export const WebSearchShorthandParametersUserLocation$outboundSchema = z.object({
    type: WebSearchShorthandParametersType$outboundSchema.optional(),
    city: z.string().optional(),
    region: z.string().optional(),
    country: z.string().optional(),
    timezone: z.string().optional(),
});
export function webSearchShorthandParametersUserLocationToJSON(webSearchShorthandParametersUserLocation) {
    return JSON.stringify(WebSearchShorthandParametersUserLocation$outboundSchema.parse(webSearchShorthandParametersUserLocation));
}
/** @internal */
export const WebSearchShorthandParameters$outboundSchema = z.object({
    engine: WebSearchShorthandParametersEngine$outboundSchema.optional(),
    maxResults: z.number().optional(),
    maxTotalResults: z.number().optional(),
    searchContextSize: WebSearchShorthandParametersSearchContextSize$outboundSchema.optional(),
    userLocation: z.lazy(() => WebSearchShorthandParametersUserLocation$outboundSchema).optional(),
    allowedDomains: z.array(z.string()).optional(),
    excludedDomains: z.array(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        maxResults: "max_results",
        maxTotalResults: "max_total_results",
        searchContextSize: "search_context_size",
        userLocation: "user_location",
        allowedDomains: "allowed_domains",
        excludedDomains: "excluded_domains",
    });
});
export function webSearchShorthandParametersToJSON(webSearchShorthandParameters) {
    return JSON.stringify(WebSearchShorthandParameters$outboundSchema.parse(webSearchShorthandParameters));
}
/** @internal */
export const WebSearchShorthand$outboundSchema = z.object({
    type: WebSearchShorthandType$outboundSchema,
    engine: WebSearchShorthandEngine$outboundSchema.optional(),
    maxResults: z.number().optional(),
    maxTotalResults: z.number().optional(),
    searchContextSize: WebSearchShorthandSearchContextSize$outboundSchema
        .optional(),
    userLocation: z.lazy(() => WebSearchShorthandUserLocation$outboundSchema)
        .optional(),
    allowedDomains: z.array(z.string()).optional(),
    excludedDomains: z.array(z.string()).optional(),
    parameters: z.lazy(() => WebSearchShorthandParameters$outboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        maxResults: "max_results",
        maxTotalResults: "max_total_results",
        searchContextSize: "search_context_size",
        userLocation: "user_location",
        allowedDomains: "allowed_domains",
        excludedDomains: "excluded_domains",
    });
});
export function webSearchShorthandToJSON(webSearchShorthand) {
    return JSON.stringify(WebSearchShorthand$outboundSchema.parse(webSearchShorthand));
}
