// @openrouter/sdk/esm/models/websearchservertool.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 478da2f8e455
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import * as openEnums from "../types/enums.js";
/**
 * Which search engine to use. "auto" (default) uses native if the provider supports it, otherwise Exa. "native" forces the provider's built-in search. "exa" forces the Exa search API.
 */
export const WebSearchServerToolEngine = {
    Auto: "auto",
    Native: "native",
    Exa: "exa",
};
/**
 * How much context to retrieve per result. Defaults to medium (15000 chars).
 */
export const WebSearchServerToolSearchContextSize = {
    Low: "low",
    Medium: "medium",
    High: "high",
};
export const WebSearchServerToolParametersType = {
    Approximate: "approximate",
};
/** @internal */
export const WebSearchServerToolEngine$outboundSchema = openEnums.outboundSchema(WebSearchServerToolEngine);
/** @internal */
export const WebSearchServerToolSearchContextSize$outboundSchema = openEnums.outboundSchema(WebSearchServerToolSearchContextSize);
/** @internal */
export const WebSearchServerToolParametersType$outboundSchema = z.enum(WebSearchServerToolParametersType);
/** @internal */
export const WebSearchServerToolUserLocation$outboundSchema = z.object({
    type: WebSearchServerToolParametersType$outboundSchema.optional(),
    city: z.string().optional(),
    region: z.string().optional(),
    country: z.string().optional(),
    timezone: z.string().optional(),
});
export function webSearchServerToolUserLocationToJSON(webSearchServerToolUserLocation) {
    return JSON.stringify(WebSearchServerToolUserLocation$outboundSchema.parse(webSearchServerToolUserLocation));
}
/** @internal */
export const WebSearchServerToolParameters$outboundSchema = z.object({
    engine: WebSearchServerToolEngine$outboundSchema.optional(),
    maxResults: z.number().optional(),
    maxTotalResults: z.number().optional(),
    searchContextSize: WebSearchServerToolSearchContextSize$outboundSchema
        .optional(),
    userLocation: z.lazy(() => WebSearchServerToolUserLocation$outboundSchema)
        .optional(),
    allowedDomains: z.array(z.string()).optional(),
    excludedDomains: z.array(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        maxResults: "max_results",
        maxTotalResults: "max_total_results",
        searchContextSize: "search_context_size",
        userLocation: "user_location",
        allowedDomains: "allowed_domains",
        excludedDomains: "excluded_domains",
    });
});
export function webSearchServerToolParametersToJSON(webSearchServerToolParameters) {
    return JSON.stringify(WebSearchServerToolParameters$outboundSchema.parse(webSearchServerToolParameters));
}
/** @internal */
export const WebSearchServerTool$outboundSchema = z.object({
    type: z.literal("openrouter:web_search"),
    parameters: z.lazy(() => WebSearchServerToolParameters$outboundSchema)
        .optional(),
});
export function webSearchServerToolToJSON(webSearchServerTool) {
    return JSON.stringify(WebSearchServerTool$outboundSchema.parse(webSearchServerTool));
}
